<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Basic Use of ceRNAnetsim • ceRNAnetsim</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Basic Use of ceRNAnetsim">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-84499524-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-84499524-2');
</script>
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ceRNAnetsim</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.22</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-book"></span>
     
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/basic_usage.html">What does ceRNAnetsim do?</a>
    </li>
    <li>
      <a href="../articles/convenient_iteration.html">How to calculate proper number of iterations for simulation of the network</a>
    </li>
    <li>
      <a href="../articles/mirtarbase_example.html">How does the system behave in mirtarbase dataset without interaction factors?</a>
    </li>
    <li>
      <a href="../articles/auxiliary_commands.html">Helper Commands</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/selcenari/ceRNAnetsim">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Basic Use of ceRNAnetsim</h1>
                        <h4 class="author">Selcen Ari</h4>
            
            <h4 class="date">2019-11-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/selcenari/ceRNAnetsim/blob/master/vignettes/basic_usage.Rmd"><code>vignettes/basic_usage.Rmd</code></a></small>
      <div class="hidden name"><code>basic_usage.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>1. Introduction</h1>
<p>This vignette demonstrates how to analyse miRNA:Competing interactions via <code>ceRNAnetsim</code> package. The perturbations in the miRNA:target interactions are handled step by step in <code>ceRNAnetsim</code>. The package calculates and simulates regulation of miRNA:competing RNA interactions based on amounts of miRNA and the targets and interaction factors.</p>
<p>The <code>ceRNAnetsim</code> works by executing following steps:</p>
<ul>
<li>The input dataset is prepared for analysis. The first variable is arranged as competing and the second as miRNA.</li>
<li>The dataset is primed with <code><a href="../reference/priming_graph.html">priming_graph()</a></code> function so that it’s converted into a graph. This function makes calculations that are depended on miRNA amount, target (competing) amount and the interaction factors. It determines the efficiency of miRNA to each target and saves that values as edge data. All calculations are performed in edge data. After that, results of calculations are used in node data.</li>
<li>Now, the change in expression level of either miRNA or competing RNA can be introduced with <code><a href="../reference/update_variables.html">update_variables()</a></code> or <code><a href="../reference/update_how.html">update_how()</a></code> functions.</li>
<li>The calculated values in edge are carried to node data through a helper function <code><a href="../reference/update_nodes.html">update_nodes()</a></code>.</li>
<li>The change is perceived as a trigger and network-wise calculations are performed.</li>
<li>The trigger is used for simulation of regulations in <code><a href="../reference/simulate.html">simulate()</a></code> or <code><a href="../reference/simulate_vis.html">simulate_vis()</a></code>.</li>
<li>These processes can generate various outputs such as graphs or graph objects.</li>
</ul>
<p>The workflow of <code>ceRNAnetsim</code> are shown as following:</p>
<p><img src="gifs/model_vignette.png" width="407" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ceRNAnetsim)</a></code></pre></div>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>2. Installation</h1>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">    <span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">"ceRNAnetsim"</span>)</a></code></pre></div>
</div>
<div id="about-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#about-the-data" class="anchor"></a>3. About the data</h1>
<p>Below is the minimal data that can be used with ceRNAnetsim.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"minsamp"</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">minsamp <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="st">  </span><span class="kw">select</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt;   competing miRNA Competing_expression miRNA_expression</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt; 1     Gene1  Mir1                10000             1000</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt; 2     Gene2  Mir1                10000             1000</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt; 3     Gene3  Mir1                 5000             1000</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt; 4     Gene4  Mir1                10000             1000</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt; 5     Gene4  Mir2                10000             2000</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; 6     Gene5  Mir2                 5000             2000</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt; 7     Gene6  Mir2                10000             2000</span></a></code></pre></div>
<p>The table is actually constructed by merging three different tables:</p>
<ul>
<li>gene expression data</li>
<li>miRNA expression data</li>
<li>miRNA:gene pairs (might <strong>optionally</strong> contain data about the interaction)</li>
</ul>
<p>So, the <code>basic_data</code> table is constructed by merging following tables:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"minsamp"</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">minsamp <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="st">  </span><span class="kw">select</span>(competing, Competing_expression) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="st">  </span><span class="kw">distinct</span>() -&gt;<span class="st"> </span>gene_expression</a></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">gene_expression</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co">#&gt;   competing Competing_expression</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; 1     Gene1                10000</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; 2     Gene2                10000</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt; 3     Gene3                 5000</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt; 4     Gene4                10000</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt; 5     Gene5                 5000</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; 6     Gene6                10000</span></a></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">minsamp <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="st">  </span><span class="kw">select</span>(miRNA, miRNA_expression) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="st">  </span><span class="kw">distinct</span>() -&gt;<span class="st"> </span>mirna_expression</a></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">mirna_expression</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt;   miRNA miRNA_expression</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; 1  Mir1             1000</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; 2  Mir2             2000</span></a></code></pre></div>
<p>Third table should contain miRNA:gene interactions per row. The <code>ceRNAnetsim</code> will assume first column contains competing RNA names and second column to be miRNA names. If the order is different the user should indicate column names accordingly.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">minsamp <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="st">  </span><span class="kw">select</span>(competing, miRNA) -&gt;<span class="st"> </span>interaction_simple</a></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">interaction_simple</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#&gt;   competing miRNA</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#&gt; 1     Gene1  Mir1</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt; 2     Gene2  Mir1</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt; 3     Gene3  Mir1</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt; 4     Gene4  Mir1</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt; 5     Gene4  Mir2</span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt; 6     Gene5  Mir2</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt; 7     Gene6  Mir2</span></a></code></pre></div>
<p>The three tables can be joined in R (as shown below) or elsewhere to have interaction and expression data altogether in expected format.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">interaction_simple <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="st">  </span><span class="kw">inner_join</span>(gene_expression, <span class="dt">by =</span> <span class="st">"competing"</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="st">  </span><span class="kw">inner_join</span>(mirna_expression, <span class="st">"miRNA"</span>) -&gt;<span class="st"> </span>basic_data</a>
<a class="sourceLine" id="cb10-4" data-line-number="4"></a>
<a class="sourceLine" id="cb10-5" data-line-number="5">basic_data</a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#&gt;   competing miRNA Competing_expression miRNA_expression</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt; 1     Gene1  Mir1                10000             1000</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt; 2     Gene2  Mir1                10000             1000</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="co">#&gt; 3     Gene3  Mir1                 5000             1000</span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="co">#&gt; 4     Gene4  Mir1                10000             1000</span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="co">#&gt; 5     Gene4  Mir2                10000             2000</span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="co">#&gt; 6     Gene5  Mir2                 5000             2000</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="co">#&gt; 7     Gene6  Mir2                10000             2000</span></a></code></pre></div>
</div>
<div id="simulation-via-expression-values-of-mirnas-and-genes-in-minsamp-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#simulation-via-expression-values-of-mirnas-and-genes-in-minsamp-dataset" class="anchor"></a>4. Simulation via expression values of miRNAs and genes in <code>minsamp</code> dataset</h1>
<p>ceRNAnetsim processes your dataset as graph object and simulates competing behaviours of targets when steady-state is perturbed via expression level changes in one or more genes. Let’s go over three steps:</p>
<div id="handle-basic-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#handle-basic-dataset" class="anchor"></a>4.1. Handle basic dataset</h2>
<p>In first step, the expression and interaction table is converted into graph/network. <code>tidygraph</code> is used importing the data thus both node and edge data are accessible as tables if needed. <code>priming_graph</code> generates many columns in edge/node table which are mostly for internal use.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co">#Convertion of dataset to graph.</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(basic_data, <span class="dt">competing_count =</span> Competing_expression, <span class="dt">miRNA_count =</span>miRNA_expression)</a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="co">#&gt; Warning in priming_graph(basic_data, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="co">#&gt; # A rooted tree</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="co">#&gt;   name  type      node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="co">#&gt; 1 Gene1 Competing       1         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="co">#&gt; 2 Gene2 Competing       2         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="co">#&gt; 3 Gene3 Competing       3          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="co">#&gt; 4 Gene4 Competing       4         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="co">#&gt; 5 Gene5 Competing       5          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb11-17" data-line-number="17"><span class="co">#&gt; 6 Gene6 Competing       6         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb11-18" data-line-number="18"><span class="co">#&gt; # … with 2 more rows</span></a>
<a class="sourceLine" id="cb11-19" data-line-number="19"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb11-20" data-line-number="20"><span class="co">#&gt; # Edge Data: 7 x 19</span></a>
<a class="sourceLine" id="cb11-21" data-line-number="21"><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre… miRNA_expression dummy</span></a>
<a class="sourceLine" id="cb11-22" data-line-number="22"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb11-23" data-line-number="23"><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000     1</span></a>
<a class="sourceLine" id="cb11-24" data-line-number="24"><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000     1</span></a>
<a class="sourceLine" id="cb11-25" data-line-number="25"><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000     1</span></a>
<a class="sourceLine" id="cb11-26" data-line-number="26"><span class="co">#&gt; # … with 4 more rows, and 12 more variables: afff_factor &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb11-27" data-line-number="27"><span class="co">#&gt; #   degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;, comp_count_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb11-28" data-line-number="28"><span class="co">#&gt; #   comp_count_current &lt;dbl&gt;, mirna_count_list &lt;list&gt;, mirna_count_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb11-29" data-line-number="29"><span class="co">#&gt; #   mirna_count_current &lt;dbl&gt;, mirna_count_per_dep &lt;dbl&gt;, effect_current &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb11-30" data-line-number="30"><span class="co">#&gt; #   effect_pre &lt;dbl&gt;, effect_list &lt;list&gt;</span></a></code></pre></div>
</div>
<div id="trigger-a-change" class="section level2">
<h2 class="hasAnchor">
<a href="#trigger-a-change" class="anchor"></a>4.2. Trigger a change</h2>
<p><code>update_how</code> function can be used to simulate a change in the network. (If multiple chnages are aimed to be used as trigger, <code><a href="../reference/update_variables.html">update_variables()</a></code> function should be used).</p>
<p>In the example below, expression level of “Gene2” is increased to two-fold.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"></a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(basic_data, <span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb12-3" data-line-number="3">              <span class="dt">miRNA_count =</span>miRNA_expression) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"Gene2"</span>, <span class="dt">how=</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="co">#&gt; Warning in priming_graph(basic_data, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="co">#&gt; # A rooted tree</span></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="co">#&gt;   name  type      node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb12-12" data-line-number="12"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb12-13" data-line-number="13"><span class="co">#&gt; 1 Gene1 Competing       1         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb12-14" data-line-number="14"><span class="co">#&gt; 2 Gene2 Competing       2         10000     10000         20000 Up              </span></a>
<a class="sourceLine" id="cb12-15" data-line-number="15"><span class="co">#&gt; 3 Gene3 Competing       3          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb12-16" data-line-number="16"><span class="co">#&gt; 4 Gene4 Competing       4         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb12-17" data-line-number="17"><span class="co">#&gt; 5 Gene5 Competing       5          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb12-18" data-line-number="18"><span class="co">#&gt; 6 Gene6 Competing       6         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb12-19" data-line-number="19"><span class="co">#&gt; # … with 2 more rows</span></a>
<a class="sourceLine" id="cb12-20" data-line-number="20"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb12-21" data-line-number="21"><span class="co">#&gt; # Edge Data: 7 x 19</span></a>
<a class="sourceLine" id="cb12-22" data-line-number="22"><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre… miRNA_expression dummy</span></a>
<a class="sourceLine" id="cb12-23" data-line-number="23"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb12-24" data-line-number="24"><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000     1</span></a>
<a class="sourceLine" id="cb12-25" data-line-number="25"><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000     1</span></a>
<a class="sourceLine" id="cb12-26" data-line-number="26"><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000     1</span></a>
<a class="sourceLine" id="cb12-27" data-line-number="27"><span class="co">#&gt; # … with 4 more rows, and 12 more variables: afff_factor &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb12-28" data-line-number="28"><span class="co">#&gt; #   degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;, comp_count_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb12-29" data-line-number="29"><span class="co">#&gt; #   comp_count_current &lt;dbl&gt;, mirna_count_list &lt;list&gt;, mirna_count_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb12-30" data-line-number="30"><span class="co">#&gt; #   mirna_count_current &lt;dbl&gt;, mirna_count_per_dep &lt;dbl&gt;, effect_current &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb12-31" data-line-number="31"><span class="co">#&gt; #   effect_pre &lt;dbl&gt;, effect_list &lt;list&gt;</span></a></code></pre></div>
<p>You can see the current count of Gene2 node is 20000 and its change is denoted as “Up” in <code>changes_variable</code> column in node table data.</p>
</div>
<div id="simulate-the-changes-in-graph" class="section level2">
<h2 class="hasAnchor">
<a href="#simulate-the-changes-in-graph" class="anchor"></a>4.3. Simulate the changes in graph</h2>
<p>Finally, with the help of <code>simulate</code> function, the effect of expression change (<em>i.e.</em> the trigger) on overall network. The example code advances only for 5 cycles.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(basic_data, <span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb13-3" data-line-number="3">              <span class="dt">miRNA_count =</span>miRNA_expression) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"Gene2"</span>, <span class="dt">how=</span><span class="dv">2</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dt">cycle =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="co">#&gt; Warning in priming_graph(basic_data, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="co">#&gt; # A rooted tree</span></a>
<a class="sourceLine" id="cb13-10" data-line-number="10"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12"><span class="co">#&gt;   name  type      node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb13-13" data-line-number="13"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb13-14" data-line-number="14"><span class="co">#&gt; 1 Gene1 Competing       1         10000    10062.        10062. Down            </span></a>
<a class="sourceLine" id="cb13-15" data-line-number="15"><span class="co">#&gt; 2 Gene2 Competing       2         10000    19845.        19845. Down            </span></a>
<a class="sourceLine" id="cb13-16" data-line-number="16"><span class="co">#&gt; 3 Gene3 Competing       3          5000     5031.         5031. Down            </span></a>
<a class="sourceLine" id="cb13-17" data-line-number="17"><span class="co">#&gt; 4 Gene4 Competing       4         10000    10059.        10059. Up              </span></a>
<a class="sourceLine" id="cb13-18" data-line-number="18"><span class="co">#&gt; 5 Gene5 Competing       5          5000     5001.         5001. Down            </span></a>
<a class="sourceLine" id="cb13-19" data-line-number="19"><span class="co">#&gt; 6 Gene6 Competing       6         10000    10002.        10002. Down            </span></a>
<a class="sourceLine" id="cb13-20" data-line-number="20"><span class="co">#&gt; # … with 2 more rows</span></a>
<a class="sourceLine" id="cb13-21" data-line-number="21"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb13-22" data-line-number="22"><span class="co">#&gt; # Edge Data: 7 x 20</span></a>
<a class="sourceLine" id="cb13-23" data-line-number="23"><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre… miRNA_expression dummy</span></a>
<a class="sourceLine" id="cb13-24" data-line-number="24"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb13-25" data-line-number="25"><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000     1</span></a>
<a class="sourceLine" id="cb13-26" data-line-number="26"><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000     1</span></a>
<a class="sourceLine" id="cb13-27" data-line-number="27"><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000     1</span></a>
<a class="sourceLine" id="cb13-28" data-line-number="28"><span class="co">#&gt; # … with 4 more rows, and 13 more variables: afff_factor &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb13-29" data-line-number="29"><span class="co">#&gt; #   degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;, comp_count_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb13-30" data-line-number="30"><span class="co">#&gt; #   comp_count_current &lt;dbl&gt;, mirna_count_list &lt;list&gt;, mirna_count_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb13-31" data-line-number="31"><span class="co">#&gt; #   mirna_count_current &lt;dbl&gt;, mirna_count_per_dep &lt;dbl&gt;, effect_current &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb13-32" data-line-number="32"><span class="co">#&gt; #   effect_pre &lt;dbl&gt;, effect_list &lt;list&gt;, mirna_count_per_comp &lt;dbl&gt;</span></a></code></pre></div>
<p><code>count_current</code> column indicate results after 5 cycle of calculations for each node. You can see the gene expression changes after this perturbation and simulation. This table can be obtained easily at following:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(basic_data, <span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb14-2" data-line-number="2">              <span class="dt">miRNA_count =</span>miRNA_expression) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"Gene2"</span>, <span class="dt">how=</span><span class="dv">2</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dt">cycle =</span> <span class="dv">5</span>)<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="st">  </span><span class="kw">as_tibble</span>()<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="st">  </span><span class="kw">select</span>(name, initial_count, count_current)</a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="co">#&gt; Warning in priming_graph(basic_data, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="co">#&gt; # A tibble: 8 x 3</span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co">#&gt;   name  initial_count count_current</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="co">#&gt;   &lt;chr&gt;         &lt;dbl&gt;         &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="co">#&gt; 1 Gene1         10000        10062.</span></a>
<a class="sourceLine" id="cb14-12" data-line-number="12"><span class="co">#&gt; 2 Gene2         10000        19845.</span></a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="co">#&gt; 3 Gene3          5000         5031.</span></a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="co">#&gt; 4 Gene4         10000        10059.</span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="co">#&gt; 5 Gene5          5000         5001.</span></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"><span class="co">#&gt; 6 Gene6         10000        10002.</span></a>
<a class="sourceLine" id="cb14-17" data-line-number="17"><span class="co">#&gt; 7 Mir1           1000         1000 </span></a>
<a class="sourceLine" id="cb14-18" data-line-number="18"><span class="co">#&gt; 8 Mir2           2000         2000</span></a></code></pre></div>
</div>
<div id="a-special-case-knockdown" class="section level2">
<h2 class="hasAnchor">
<a href="#a-special-case-knockdown" class="anchor"></a>4.4. A special case: knockdown</h2>
<p>ceRNAnetsim also provides the simulation of gene knockdown in the network. In normal conditions, when a gene is up or down regulated, it is considered that amounts of gene transcripts change depended on interactions. But, the transcripts of the gene are not observed in the system when it is knocked down. To achieve this case, you just need to define <code>how</code> argument to 0 (zero) in <code>update_how</code> function.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"></a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(basic_data, <span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb15-3" data-line-number="3">              <span class="dt">miRNA_count =</span>miRNA_expression) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"Gene2"</span>, <span class="dt">how=</span><span class="dv">0</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dt">cycle =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="co">#&gt; Warning in priming_graph(basic_data, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb15-7" data-line-number="7"><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></a>
<a class="sourceLine" id="cb15-8" data-line-number="8"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb15-9" data-line-number="9"><span class="co">#&gt; # A rooted tree</span></a>
<a class="sourceLine" id="cb15-10" data-line-number="10"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb15-11" data-line-number="11"><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></a>
<a class="sourceLine" id="cb15-12" data-line-number="12"><span class="co">#&gt;   name  type      node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb15-13" data-line-number="13"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb15-14" data-line-number="14"><span class="co">#&gt; 1 Gene1 Competing       1         10000     9886.         9886. Up              </span></a>
<a class="sourceLine" id="cb15-15" data-line-number="15"><span class="co">#&gt; 2 Gene2 Competing       2         10000        0             0  Competing       </span></a>
<a class="sourceLine" id="cb15-16" data-line-number="16"><span class="co">#&gt; 3 Gene3 Competing       3          5000     4943.         4943. Up              </span></a>
<a class="sourceLine" id="cb15-17" data-line-number="17"><span class="co">#&gt; 4 Gene4 Competing       4         10000     9891.         9891. Down            </span></a>
<a class="sourceLine" id="cb15-18" data-line-number="18"><span class="co">#&gt; 5 Gene5 Competing       5          5000     4998.         4998. Up              </span></a>
<a class="sourceLine" id="cb15-19" data-line-number="19"><span class="co">#&gt; 6 Gene6 Competing       6         10000     9997.         9997. Up              </span></a>
<a class="sourceLine" id="cb15-20" data-line-number="20"><span class="co">#&gt; # … with 2 more rows</span></a>
<a class="sourceLine" id="cb15-21" data-line-number="21"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb15-22" data-line-number="22"><span class="co">#&gt; # Edge Data: 7 x 20</span></a>
<a class="sourceLine" id="cb15-23" data-line-number="23"><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre… miRNA_expression dummy</span></a>
<a class="sourceLine" id="cb15-24" data-line-number="24"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb15-25" data-line-number="25"><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000     1</span></a>
<a class="sourceLine" id="cb15-26" data-line-number="26"><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000     1</span></a>
<a class="sourceLine" id="cb15-27" data-line-number="27"><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000     1</span></a>
<a class="sourceLine" id="cb15-28" data-line-number="28"><span class="co">#&gt; # … with 4 more rows, and 13 more variables: afff_factor &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb15-29" data-line-number="29"><span class="co">#&gt; #   degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;, comp_count_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb15-30" data-line-number="30"><span class="co">#&gt; #   comp_count_current &lt;dbl&gt;, mirna_count_list &lt;list&gt;, mirna_count_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb15-31" data-line-number="31"><span class="co">#&gt; #   mirna_count_current &lt;dbl&gt;, mirna_count_per_dep &lt;dbl&gt;, effect_current &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb15-32" data-line-number="32"><span class="co">#&gt; #   effect_pre &lt;dbl&gt;, effect_list &lt;list&gt;, mirna_count_per_comp &lt;dbl&gt;</span></a></code></pre></div>
<p>So, if Gene2 is knocked down, there will be more miRNA (Mir1 to be exact) available for Gene1, Gene3 and Gene4, thus lowering their transcript levels. Since Gene4 is has lower expression level, we can observe minute changes in Gene5 and Gene6 levels due to more miRNA (Mir2) being available for them. These changes can be observed in <code>current_count</code> column.</p>
<p>Briefly, ceRNAnetsim utilizes the change(s) as trigger and calculates regulation of targets according to miRNA:target and target:total target ratios.</p>
</div>
</div>
<div id="simulation-via-interaction-factors-in-addition-to-expression-values-of-mirnas-and-genes-in-minsamp-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#simulation-via-interaction-factors-in-addition-to-expression-values-of-mirnas-and-genes-in-minsamp-dataset" class="anchor"></a>5. Simulation via interaction factors in addition to expression values of miRNAs and genes in <code>minsamp</code> dataset</h1>
<p>Minimal sample <code>minsamp</code> is processed with <code><a href="../reference/priming_graph.html">priming_graph()</a></code> function in first step. This provides conversion of dataset from data frame to graph object. This step comprises of:</p>
<ul>
<li>The competing elements (Genes in <em>minsamp</em>) are grouped according to the miRNAs they are associated with.</li>
<li>In graph object, the <strong>optional</strong> interaction factors are processed and graded within the groups.</li>
<li>The amounts (expressions) of miRNAs are distributed according to competing:total competing ratio.</li>
<li>miRNA efficiency in steady-state is calculated by taking into account of expression distribution and effecting factors. The <strong>optional</strong> factors might have two types of effect; 1) affinity, 2) degradation effect. Any column which has effect on affinity should be provided as a vector to <code>aff_factor</code> argument and any column that effects degradation of target RNA should be as a vector to <code>deg_factor</code> argument.</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">minsamp</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co">#&gt;   competing miRNA Competing_expression miRNA_expression seed_type region energy</span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="co">#&gt; 1     Gene1  Mir1                10000             1000      0.43   0.30    -20</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co">#&gt; 2     Gene2  Mir1                10000             1000      0.43   0.01    -15</span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="co">#&gt; 3     Gene3  Mir1                 5000             1000      0.32   0.40    -14</span></a>
<a class="sourceLine" id="cb16-6" data-line-number="6"><span class="co">#&gt; 4     Gene4  Mir1                10000             1000      0.23   0.50    -10</span></a>
<a class="sourceLine" id="cb16-7" data-line-number="7"><span class="co">#&gt; 5     Gene4  Mir2                10000             2000      0.35   0.90    -12</span></a>
<a class="sourceLine" id="cb16-8" data-line-number="8"><span class="co">#&gt; 6     Gene5  Mir2                 5000             2000      0.05   0.40    -11</span></a>
<a class="sourceLine" id="cb16-9" data-line-number="9"><span class="co">#&gt; 7     Gene6  Mir2                10000             2000      0.01   0.80    -25</span></a></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(minsamp, </a>
<a class="sourceLine" id="cb17-2" data-line-number="2">              <span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb17-3" data-line-number="3">              <span class="dt">miRNA_count =</span> miRNA_expression,</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">              <span class="dt">aff_factor =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(energy, seed_type), </a>
<a class="sourceLine" id="cb17-5" data-line-number="5">              <span class="dt">deg_factor =</span> region)</a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="co">#&gt; Warning in priming_graph(minsamp, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb17-7" data-line-number="7"><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></a>
<a class="sourceLine" id="cb17-8" data-line-number="8"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb17-9" data-line-number="9"><span class="co">#&gt; # A rooted tree</span></a>
<a class="sourceLine" id="cb17-10" data-line-number="10"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb17-11" data-line-number="11"><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></a>
<a class="sourceLine" id="cb17-12" data-line-number="12"><span class="co">#&gt;   name  type      node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb17-13" data-line-number="13"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb17-14" data-line-number="14"><span class="co">#&gt; 1 Gene1 Competing       1         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb17-15" data-line-number="15"><span class="co">#&gt; 2 Gene2 Competing       2         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb17-16" data-line-number="16"><span class="co">#&gt; 3 Gene3 Competing       3          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb17-17" data-line-number="17"><span class="co">#&gt; 4 Gene4 Competing       4         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb17-18" data-line-number="18"><span class="co">#&gt; 5 Gene5 Competing       5          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb17-19" data-line-number="19"><span class="co">#&gt; 6 Gene6 Competing       6         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb17-20" data-line-number="20"><span class="co">#&gt; # … with 2 more rows</span></a>
<a class="sourceLine" id="cb17-21" data-line-number="21"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb17-22" data-line-number="22"><span class="co">#&gt; # Edge Data: 7 x 22</span></a>
<a class="sourceLine" id="cb17-23" data-line-number="23"><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre… miRNA_expression energy</span></a>
<a class="sourceLine" id="cb17-24" data-line-number="24"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb17-25" data-line-number="25"><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000    -20</span></a>
<a class="sourceLine" id="cb17-26" data-line-number="26"><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000    -15</span></a>
<a class="sourceLine" id="cb17-27" data-line-number="27"><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000    -14</span></a>
<a class="sourceLine" id="cb17-28" data-line-number="28"><span class="co">#&gt; # … with 4 more rows, and 15 more variables: seed_type &lt;dbl&gt;, region &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb17-29" data-line-number="29"><span class="co">#&gt; #   dummy &lt;dbl&gt;, afff_factor &lt;dbl&gt;, degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb17-30" data-line-number="30"><span class="co">#&gt; #   comp_count_pre &lt;dbl&gt;, comp_count_current &lt;dbl&gt;, mirna_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb17-31" data-line-number="31"><span class="co">#&gt; #   mirna_count_pre &lt;dbl&gt;, mirna_count_current &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb17-32" data-line-number="32"><span class="co">#&gt; #   mirna_count_per_dep &lt;dbl&gt;, effect_current &lt;dbl&gt;, effect_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb17-33" data-line-number="33"><span class="co">#&gt; #   effect_list &lt;list&gt;</span></a></code></pre></div>
<p>In the processed data, the values are carried as node variables and many more columns are initialized which are to be used in subsequent steps.</p>
<div id="change-expression-level-of-one-or-more-nodes-in-the-graph" class="section level2">
<h2 class="hasAnchor">
<a href="#change-expression-level-of-one-or-more-nodes-in-the-graph" class="anchor"></a>5.1. Change expression level of one or more nodes in the graph</h2>
<p>In the steady-state, the miRNA degradation effect on gene expression is assumed to be stable (<em>i.e.</em> in equilibrium). But, if one or more nodes have altered expression level, the system tends to reach steady-state again.</p>
<p>The <code>ceRNAnetsim</code> package utilizes two methods to simulate change in expression level, <code><a href="../reference/update_how.html">update_how()</a></code> and <code><a href="../reference/update_variables.html">update_variables()</a></code> functions provide unstable state from which calculations are triggered to reach steady-state.</p>
<ul>
<li>Method 1: change expression level of single node</li>
</ul>
<p>If updating expression level of single node is desired then <code><a href="../reference/update_how.html">update_how()</a></code> function should be used. In the example below, expression level of Gene4 is increased 2-fold.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">minsamp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="st">   </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb18-3" data-line-number="3">                 <span class="dt">miRNA_count =</span> miRNA_expression,</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">                 <span class="dt">aff_factor =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(energy, seed_type), </a>
<a class="sourceLine" id="cb18-5" data-line-number="5">                 <span class="dt">deg_factor =</span> region) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb18-6" data-line-number="6"><span class="st">   </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"Gene4"</span>, <span class="dt">how =</span> <span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb18-7" data-line-number="7"><span class="st">   </span><span class="kw">activate</span>(edges)<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb18-8" data-line-number="8"><span class="st">   </span><span class="co"># following line is just for focusing on necessary</span></a>
<a class="sourceLine" id="cb18-9" data-line-number="9"><span class="st">   </span><span class="co"># columns to see the change in edge data</span></a>
<a class="sourceLine" id="cb18-10" data-line-number="10"><span class="st">   </span><span class="kw">select</span>(<span class="dv">3</span><span class="op">:</span><span class="dv">4</span>,comp_count_pre,comp_count_current)</a>
<a class="sourceLine" id="cb18-11" data-line-number="11"><span class="co">#&gt; Warning in priming_graph(., competing_count = Competing_expression, miRNA_count = miRNA_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb18-12" data-line-number="12"><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></a>
<a class="sourceLine" id="cb18-13" data-line-number="13"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb18-14" data-line-number="14"><span class="co">#&gt; # A rooted tree</span></a>
<a class="sourceLine" id="cb18-15" data-line-number="15"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb18-16" data-line-number="16"><span class="co">#&gt; # Edge Data: 7 x 6 (active)</span></a>
<a class="sourceLine" id="cb18-17" data-line-number="17"><span class="co">#&gt;    from    to Competing_name miRNA_name comp_count_pre comp_count_current</span></a>
<a class="sourceLine" id="cb18-18" data-line-number="18"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;               &lt;dbl&gt;              &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb18-19" data-line-number="19"><span class="co">#&gt; 1     1     7 Gene1          Mir1                10000              10000</span></a>
<a class="sourceLine" id="cb18-20" data-line-number="20"><span class="co">#&gt; 2     2     7 Gene2          Mir1                10000              10000</span></a>
<a class="sourceLine" id="cb18-21" data-line-number="21"><span class="co">#&gt; 3     3     7 Gene3          Mir1                 5000               5000</span></a>
<a class="sourceLine" id="cb18-22" data-line-number="22"><span class="co">#&gt; 4     4     7 Gene4          Mir1                10000              20000</span></a>
<a class="sourceLine" id="cb18-23" data-line-number="23"><span class="co">#&gt; 5     4     8 Gene4          Mir2                10000              20000</span></a>
<a class="sourceLine" id="cb18-24" data-line-number="24"><span class="co">#&gt; 6     5     8 Gene5          Mir2                 5000               5000</span></a>
<a class="sourceLine" id="cb18-25" data-line-number="25"><span class="co">#&gt; # … with 1 more row</span></a>
<a class="sourceLine" id="cb18-26" data-line-number="26"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb18-27" data-line-number="27"><span class="co">#&gt; # Node Data: 8 x 7</span></a>
<a class="sourceLine" id="cb18-28" data-line-number="28"><span class="co">#&gt;   name  type      node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb18-29" data-line-number="29"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb18-30" data-line-number="30"><span class="co">#&gt; 1 Gene1 Competing       1         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb18-31" data-line-number="31"><span class="co">#&gt; 2 Gene2 Competing       2         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb18-32" data-line-number="32"><span class="co">#&gt; 3 Gene3 Competing       3          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb18-33" data-line-number="33"><span class="co">#&gt; # … with 5 more rows</span></a></code></pre></div>
<ul>
<li>Method 2: update expression level of all nodes</li>
</ul>
<p>The <code><a href="../reference/update_variables.html">update_variables()</a></code> function uses an external dataset which has number of rows equal to number of nodes in graph. The external dataset might include changed and unchanged expression values for each node.</p>
<p>Load the <code>new_count</code> dataset (provided with package sample data) in which expression level of Gene2 is increased 2 fold (from 10,000 to 20,000). Note that variables of the dataset included updated variables must be named as “Competing”, “miRNA”, “miRNA_count” and “Competing_count”.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(new_counts) </a>
<a class="sourceLine" id="cb19-2" data-line-number="2">new_counts</a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="co">#&gt;   Competing miRNA Competing_count miRNA_count</span></a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="co">#&gt; 1     Gene1  Mir1           10000        1000</span></a>
<a class="sourceLine" id="cb19-5" data-line-number="5"><span class="co">#&gt; 2     Gene2  Mir1           20000        1000</span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="co">#&gt; 3     Gene3  Mir1            5000        1000</span></a>
<a class="sourceLine" id="cb19-7" data-line-number="7"><span class="co">#&gt; 4     Gene4  Mir1           10000        1000</span></a>
<a class="sourceLine" id="cb19-8" data-line-number="8"><span class="co">#&gt; 5     Gene4  Mir2           10000        2000</span></a>
<a class="sourceLine" id="cb19-9" data-line-number="9"><span class="co">#&gt; 6     Gene5  Mir2            5000        2000</span></a>
<a class="sourceLine" id="cb19-10" data-line-number="10"><span class="co">#&gt; 7     Gene6  Mir2           10000        2000</span></a></code></pre></div>
<p><code><a href="../reference/update_variables.html">update_variables()</a></code> function replaces the existing expression values with new values. The function checks for updates in all rows after importing expression values, thus it’s possible to introduce multiple changes at once.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">minsamp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="st">   </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb20-3" data-line-number="3">                 <span class="dt">miRNA_count =</span> miRNA_expression,</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">                 <span class="dt">aff_factor =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(energy, seed_type), </a>
<a class="sourceLine" id="cb20-5" data-line-number="5">                 <span class="dt">deg_factor =</span> region) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-6" data-line-number="6"><span class="st">   </span><span class="kw"><a href="../reference/update_variables.html">update_variables</a></span>(<span class="dt">current_counts =</span> new_counts)</a>
<a class="sourceLine" id="cb20-7" data-line-number="7"><span class="co">#&gt; Warning in priming_graph(., competing_count = Competing_expression, miRNA_count = miRNA_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb20-8" data-line-number="8"><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></a>
<a class="sourceLine" id="cb20-9" data-line-number="9"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb20-10" data-line-number="10"><span class="co">#&gt; # A rooted tree</span></a>
<a class="sourceLine" id="cb20-11" data-line-number="11"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb20-12" data-line-number="12"><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></a>
<a class="sourceLine" id="cb20-13" data-line-number="13"><span class="co">#&gt;   name  type      node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb20-14" data-line-number="14"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb20-15" data-line-number="15"><span class="co">#&gt; 1 Gene1 Competing       1         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb20-16" data-line-number="16"><span class="co">#&gt; 2 Gene2 Competing       2         10000     10000         20000 Up              </span></a>
<a class="sourceLine" id="cb20-17" data-line-number="17"><span class="co">#&gt; 3 Gene3 Competing       3          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb20-18" data-line-number="18"><span class="co">#&gt; 4 Gene4 Competing       4         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb20-19" data-line-number="19"><span class="co">#&gt; 5 Gene5 Competing       5          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb20-20" data-line-number="20"><span class="co">#&gt; 6 Gene6 Competing       6         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb20-21" data-line-number="21"><span class="co">#&gt; # … with 2 more rows</span></a>
<a class="sourceLine" id="cb20-22" data-line-number="22"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb20-23" data-line-number="23"><span class="co">#&gt; # Edge Data: 7 x 22</span></a>
<a class="sourceLine" id="cb20-24" data-line-number="24"><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre… miRNA_expression energy</span></a>
<a class="sourceLine" id="cb20-25" data-line-number="25"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb20-26" data-line-number="26"><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000    -20</span></a>
<a class="sourceLine" id="cb20-27" data-line-number="27"><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000    -15</span></a>
<a class="sourceLine" id="cb20-28" data-line-number="28"><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000    -14</span></a>
<a class="sourceLine" id="cb20-29" data-line-number="29"><span class="co">#&gt; # … with 4 more rows, and 15 more variables: seed_type &lt;dbl&gt;, region &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb20-30" data-line-number="30"><span class="co">#&gt; #   dummy &lt;dbl&gt;, afff_factor &lt;dbl&gt;, degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb20-31" data-line-number="31"><span class="co">#&gt; #   comp_count_pre &lt;dbl&gt;, comp_count_current &lt;dbl&gt;, mirna_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb20-32" data-line-number="32"><span class="co">#&gt; #   mirna_count_pre &lt;dbl&gt;, mirna_count_current &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb20-33" data-line-number="33"><span class="co">#&gt; #   mirna_count_per_dep &lt;dbl&gt;, effect_current &lt;dbl&gt;, effect_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb20-34" data-line-number="34"><span class="co">#&gt; #   effect_list &lt;list&gt;</span></a></code></pre></div>
</div>
<div id="update-the-node-variables-with-edge-variables-" class="section level2">
<h2 class="hasAnchor">
<a href="#update-the-node-variables-with-edge-variables-" class="anchor"></a>5.2. Update the node variables with edge variables.</h2>
<p>The functions <code><a href="../reference/update_variables.html">update_variables()</a></code> and <code><a href="../reference/update_how.html">update_how()</a></code> updates edge data. In these functions, <code><a href="../reference/update_nodes.html">update_nodes()</a></code> function is applied in order to reflect changes in edge data over to node data. In other words, if there’s a change in edge data, nodes can be updated accordingly with <code><a href="../reference/update_nodes.html">update_nodes()</a></code> function.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">minsamp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb21-3" data-line-number="3">                <span class="dt">miRNA_count =</span> miRNA_expression,</a>
<a class="sourceLine" id="cb21-4" data-line-number="4">                <span class="dt">aff_factor =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(energy, seed_type), </a>
<a class="sourceLine" id="cb21-5" data-line-number="5">                <span class="dt">deg_factor =</span> region) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="st">"Gene4"</span>, <span class="dt">how =</span> <span class="dv">2</span>) </a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="co">#&gt; Warning in priming_graph(., competing_count = Competing_expression, miRNA_count = miRNA_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></a>
<a class="sourceLine" id="cb21-9" data-line-number="9"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb21-10" data-line-number="10"><span class="co">#&gt; # A rooted tree</span></a>
<a class="sourceLine" id="cb21-11" data-line-number="11"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb21-12" data-line-number="12"><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></a>
<a class="sourceLine" id="cb21-13" data-line-number="13"><span class="co">#&gt;   name  type      node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb21-14" data-line-number="14"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb21-15" data-line-number="15"><span class="co">#&gt; 1 Gene1 Competing       1         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb21-16" data-line-number="16"><span class="co">#&gt; 2 Gene2 Competing       2         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb21-17" data-line-number="17"><span class="co">#&gt; 3 Gene3 Competing       3          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb21-18" data-line-number="18"><span class="co">#&gt; 4 Gene4 Competing       4         10000     10000         20000 Up              </span></a>
<a class="sourceLine" id="cb21-19" data-line-number="19"><span class="co">#&gt; 5 Gene5 Competing       5          5000      5000          5000 Competing       </span></a>
<a class="sourceLine" id="cb21-20" data-line-number="20"><span class="co">#&gt; 6 Gene6 Competing       6         10000     10000         10000 Competing       </span></a>
<a class="sourceLine" id="cb21-21" data-line-number="21"><span class="co">#&gt; # … with 2 more rows</span></a>
<a class="sourceLine" id="cb21-22" data-line-number="22"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb21-23" data-line-number="23"><span class="co">#&gt; # Edge Data: 7 x 22</span></a>
<a class="sourceLine" id="cb21-24" data-line-number="24"><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre… miRNA_expression energy</span></a>
<a class="sourceLine" id="cb21-25" data-line-number="25"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb21-26" data-line-number="26"><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000    -20</span></a>
<a class="sourceLine" id="cb21-27" data-line-number="27"><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000    -15</span></a>
<a class="sourceLine" id="cb21-28" data-line-number="28"><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000    -14</span></a>
<a class="sourceLine" id="cb21-29" data-line-number="29"><span class="co">#&gt; # … with 4 more rows, and 15 more variables: seed_type &lt;dbl&gt;, region &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb21-30" data-line-number="30"><span class="co">#&gt; #   dummy &lt;dbl&gt;, afff_factor &lt;dbl&gt;, degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb21-31" data-line-number="31"><span class="co">#&gt; #   comp_count_pre &lt;dbl&gt;, comp_count_current &lt;dbl&gt;, mirna_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb21-32" data-line-number="32"><span class="co">#&gt; #   mirna_count_pre &lt;dbl&gt;, mirna_count_current &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb21-33" data-line-number="33"><span class="co">#&gt; #   mirna_count_per_dep &lt;dbl&gt;, effect_current &lt;dbl&gt;, effect_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb21-34" data-line-number="34"><span class="co">#&gt; #   effect_list &lt;list&gt;</span></a>
<a class="sourceLine" id="cb21-35" data-line-number="35"></a>
<a class="sourceLine" id="cb21-36" data-line-number="36"><span class="co"># OR</span></a>
<a class="sourceLine" id="cb21-37" data-line-number="37"><span class="co"># minsamp %&gt;%</span></a>
<a class="sourceLine" id="cb21-38" data-line-number="38"><span class="co">#   priming_graph(competing_count = Competing_expression, </span></a>
<a class="sourceLine" id="cb21-39" data-line-number="39"><span class="co">#                 miRNA_count = miRNA_expression,</span></a>
<a class="sourceLine" id="cb21-40" data-line-number="40"><span class="co">#                 aff_factor = c(energy, seed_type), </span></a>
<a class="sourceLine" id="cb21-41" data-line-number="41"><span class="co">#                 deg_factor = region) %&gt;%</span></a>
<a class="sourceLine" id="cb21-42" data-line-number="42"><span class="co">#   update_variables(current_counts = new_counts) </span></a></code></pre></div>
<p><img src="gifs/small_sample_chunk10.png" width="334" style="display: block; margin: auto;"></p>
</div>
<div id="simulate-the-model" class="section level2">
<h2 class="hasAnchor">
<a href="#simulate-the-model" class="anchor"></a>5.3. Simulate the model</h2>
<p>Change in expression level of one or more nodes will trigger a perturbation in the system which will effect neighboring miRNA:target interactions. The effect will propagate and iterate over until it reaches steady-state.</p>
<p>With <code><a href="../reference/simulate.html">simulate()</a></code> function the changes in the system, are calculated iteratively. For example, in the example below, simulation will proceed ten cycles only. In simulation of the regulation, the important argument is <code>threshold</code> which provides to specify absolute minimum amount of change required to be considered changed element as up or down.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">minsamp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb22-3" data-line-number="3">                <span class="dt">miRNA_count =</span> miRNA_expression,</a>
<a class="sourceLine" id="cb22-4" data-line-number="4">                <span class="dt">aff_factor =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(energy, seed_type), </a>
<a class="sourceLine" id="cb22-5" data-line-number="5">                <span class="dt">deg_factor =</span> region) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="st">"Gene4"</span>, <span class="dt">how =</span> <span class="dv">2</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dt">cycle=</span><span class="dv">10</span>) <span class="co">#threshold with default 0.</span></a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="co">#&gt; Warning in priming_graph(., competing_count = Competing_expression, miRNA_count = miRNA_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></a>
<a class="sourceLine" id="cb22-10" data-line-number="10"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb22-11" data-line-number="11"><span class="co">#&gt; # A rooted tree</span></a>
<a class="sourceLine" id="cb22-12" data-line-number="12"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb22-13" data-line-number="13"><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></a>
<a class="sourceLine" id="cb22-14" data-line-number="14"><span class="co">#&gt;   name  type      node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb22-15" data-line-number="15"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb22-16" data-line-number="16"><span class="co">#&gt; 1 Gene1 Competing       1         10000    10027.        10027. Competing       </span></a>
<a class="sourceLine" id="cb22-17" data-line-number="17"><span class="co">#&gt; 2 Gene2 Competing       2         10000    10001.        10001. Competing       </span></a>
<a class="sourceLine" id="cb22-18" data-line-number="18"><span class="co">#&gt; 3 Gene3 Competing       3          5000     5009.         5009. Competing       </span></a>
<a class="sourceLine" id="cb22-19" data-line-number="19"><span class="co">#&gt; 4 Gene4 Competing       4         10000    19806.        19806. Competing       </span></a>
<a class="sourceLine" id="cb22-20" data-line-number="20"><span class="co">#&gt; 5 Gene5 Competing       5          5000     5024.         5024. Competing       </span></a>
<a class="sourceLine" id="cb22-21" data-line-number="21"><span class="co">#&gt; 6 Gene6 Competing       6         10000    10044.        10044. Competing       </span></a>
<a class="sourceLine" id="cb22-22" data-line-number="22"><span class="co">#&gt; # … with 2 more rows</span></a>
<a class="sourceLine" id="cb22-23" data-line-number="23"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb22-24" data-line-number="24"><span class="co">#&gt; # Edge Data: 7 x 23</span></a>
<a class="sourceLine" id="cb22-25" data-line-number="25"><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre… miRNA_expression energy</span></a>
<a class="sourceLine" id="cb22-26" data-line-number="26"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb22-27" data-line-number="27"><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000    -20</span></a>
<a class="sourceLine" id="cb22-28" data-line-number="28"><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000    -15</span></a>
<a class="sourceLine" id="cb22-29" data-line-number="29"><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000    -14</span></a>
<a class="sourceLine" id="cb22-30" data-line-number="30"><span class="co">#&gt; # … with 4 more rows, and 16 more variables: seed_type &lt;dbl&gt;, region &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb22-31" data-line-number="31"><span class="co">#&gt; #   dummy &lt;dbl&gt;, afff_factor &lt;dbl&gt;, degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb22-32" data-line-number="32"><span class="co">#&gt; #   comp_count_pre &lt;dbl&gt;, comp_count_current &lt;dbl&gt;, mirna_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb22-33" data-line-number="33"><span class="co">#&gt; #   mirna_count_pre &lt;dbl&gt;, mirna_count_current &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb22-34" data-line-number="34"><span class="co">#&gt; #   mirna_count_per_dep &lt;dbl&gt;, effect_current &lt;dbl&gt;, effect_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb22-35" data-line-number="35"><span class="co">#&gt; #   effect_list &lt;list&gt;, mirna_count_per_comp &lt;dbl&gt;</span></a></code></pre></div>
<p><code><a href="../reference/simulate.html">simulate()</a></code> saves the expression level of previous iterations in list columns in edge data. The changes in expression level throughout the simulate cycles are accessible with standard <code>dplyr</code> functions. For example:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">minsamp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="st">   </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb23-3" data-line-number="3">                 <span class="dt">miRNA_count =</span> miRNA_expression,</a>
<a class="sourceLine" id="cb23-4" data-line-number="4">                 <span class="dt">aff_factor =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(energy, seed_type), </a>
<a class="sourceLine" id="cb23-5" data-line-number="5">                 <span class="dt">deg_factor =</span> region) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="st">   </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="st">"Gene4"</span>, <span class="dt">how =</span> <span class="dv">2</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="st">   </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dt">cycle=</span><span class="dv">10</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="st">   </span><span class="kw">activate</span>(edges) <span class="op">%&gt;%</span><span class="st">       </span><span class="co">#from tidygraph package </span></a>
<a class="sourceLine" id="cb23-9" data-line-number="9"><span class="st">   </span><span class="kw">select</span>(comp_count_list, mirna_count_list) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-10" data-line-number="10"><span class="st">   </span><span class="kw">as_tibble</span>()  </a>
<a class="sourceLine" id="cb23-11" data-line-number="11"><span class="co">#&gt; Warning in priming_graph(., competing_count = Competing_expression, miRNA_count = miRNA_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb23-12" data-line-number="12"><span class="co">#&gt; # A tibble: 7 x 4</span></a>
<a class="sourceLine" id="cb23-13" data-line-number="13"><span class="co">#&gt;    from    to comp_count_list mirna_count_list</span></a>
<a class="sourceLine" id="cb23-14" data-line-number="14"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;list&gt;          &lt;list&gt;          </span></a>
<a class="sourceLine" id="cb23-15" data-line-number="15"><span class="co">#&gt; 1     1     7 &lt;dbl [11]&gt;      &lt;dbl [11]&gt;      </span></a>
<a class="sourceLine" id="cb23-16" data-line-number="16"><span class="co">#&gt; 2     2     7 &lt;dbl [11]&gt;      &lt;dbl [11]&gt;      </span></a>
<a class="sourceLine" id="cb23-17" data-line-number="17"><span class="co">#&gt; 3     3     7 &lt;dbl [11]&gt;      &lt;dbl [11]&gt;      </span></a>
<a class="sourceLine" id="cb23-18" data-line-number="18"><span class="co">#&gt; 4     4     7 &lt;dbl [11]&gt;      &lt;dbl [11]&gt;      </span></a>
<a class="sourceLine" id="cb23-19" data-line-number="19"><span class="co">#&gt; 5     4     8 &lt;dbl [11]&gt;      &lt;dbl [11]&gt;      </span></a>
<a class="sourceLine" id="cb23-20" data-line-number="20"><span class="co">#&gt; 6     5     8 &lt;dbl [11]&gt;      &lt;dbl [11]&gt;      </span></a>
<a class="sourceLine" id="cb23-21" data-line-number="21"><span class="co">#&gt; 7     6     8 &lt;dbl [11]&gt;      &lt;dbl [11]&gt;</span></a></code></pre></div>
<p>Here, <code>comp_count_list</code> and <code>mirna_count_list</code> are list-columns which track changes in both competing RNA and miRNA levels. In the sample above, “Gene4” has initial expression level of 10000 (after trigger, it’s initial expression is 20000) and reached level of 19806 at 9th cycle (count_pre) and also stayed at 19806 in last cycle (count_current). The full history of expression level for Gene4 is as follows:</p>
<pre><code>#&gt;  [1] 10000 19803 19806 19806 19806 19806 19806 19806 19806 19806 19806</code></pre>
<p>Actually, Gene4 seems like reached steady-state in iteration 4. But, this approach is sensitive to even small decimal numbers. So, threshold argument could be used to ignore very small decimal numbers. With a threshold value the system can reach steady-state early, like following.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">minsamp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb25-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb25-3" data-line-number="3">                <span class="dt">miRNA_count =</span> miRNA_expression,</a>
<a class="sourceLine" id="cb25-4" data-line-number="4">                <span class="dt">aff_factor =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(energy, seed_type), </a>
<a class="sourceLine" id="cb25-5" data-line-number="5">                <span class="dt">deg_factor =</span> region) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb25-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="st">"Gene4"</span>, <span class="dt">how =</span> <span class="dv">2</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb25-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dt">cycle=</span><span class="dv">3</span>, <span class="dt">threshold =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb25-8" data-line-number="8"><span class="co">#&gt; Warning in priming_graph(., competing_count = Competing_expression, miRNA_count = miRNA_expression, : First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb25-9" data-line-number="9"><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></a>
<a class="sourceLine" id="cb25-10" data-line-number="10"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb25-11" data-line-number="11"><span class="co">#&gt; # A rooted tree</span></a>
<a class="sourceLine" id="cb25-12" data-line-number="12"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb25-13" data-line-number="13"><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></a>
<a class="sourceLine" id="cb25-14" data-line-number="14"><span class="co">#&gt;   name  type      node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb25-15" data-line-number="15"><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;       &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb25-16" data-line-number="16"><span class="co">#&gt; 1 Gene1 Competing       1         10000    10027.        10027. Competing       </span></a>
<a class="sourceLine" id="cb25-17" data-line-number="17"><span class="co">#&gt; 2 Gene2 Competing       2         10000    10001.        10001. Competing       </span></a>
<a class="sourceLine" id="cb25-18" data-line-number="18"><span class="co">#&gt; 3 Gene3 Competing       3          5000     5009.         5009. Competing       </span></a>
<a class="sourceLine" id="cb25-19" data-line-number="19"><span class="co">#&gt; 4 Gene4 Competing       4         10000    19806.        19806. Competing       </span></a>
<a class="sourceLine" id="cb25-20" data-line-number="20"><span class="co">#&gt; 5 Gene5 Competing       5          5000     5024.         5024. Competing       </span></a>
<a class="sourceLine" id="cb25-21" data-line-number="21"><span class="co">#&gt; 6 Gene6 Competing       6         10000    10044.        10044. Competing       </span></a>
<a class="sourceLine" id="cb25-22" data-line-number="22"><span class="co">#&gt; # … with 2 more rows</span></a>
<a class="sourceLine" id="cb25-23" data-line-number="23"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb25-24" data-line-number="24"><span class="co">#&gt; # Edge Data: 7 x 23</span></a>
<a class="sourceLine" id="cb25-25" data-line-number="25"><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre… miRNA_expression energy</span></a>
<a class="sourceLine" id="cb25-26" data-line-number="26"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb25-27" data-line-number="27"><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000    -20</span></a>
<a class="sourceLine" id="cb25-28" data-line-number="28"><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000    -15</span></a>
<a class="sourceLine" id="cb25-29" data-line-number="29"><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000    -14</span></a>
<a class="sourceLine" id="cb25-30" data-line-number="30"><span class="co">#&gt; # … with 4 more rows, and 16 more variables: seed_type &lt;dbl&gt;, region &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb25-31" data-line-number="31"><span class="co">#&gt; #   dummy &lt;dbl&gt;, afff_factor &lt;dbl&gt;, degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb25-32" data-line-number="32"><span class="co">#&gt; #   comp_count_pre &lt;dbl&gt;, comp_count_current &lt;dbl&gt;, mirna_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb25-33" data-line-number="33"><span class="co">#&gt; #   mirna_count_pre &lt;dbl&gt;, mirna_count_current &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb25-34" data-line-number="34"><span class="co">#&gt; #   mirna_count_per_dep &lt;dbl&gt;, effect_current &lt;dbl&gt;, effect_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb25-35" data-line-number="35"><span class="co">#&gt; #   effect_list &lt;list&gt;, mirna_count_per_comp &lt;dbl&gt;</span></a></code></pre></div>
</div>
<div id="visualisation-of-the-graph" class="section level2">
<h2 class="hasAnchor">
<a href="#visualisation-of-the-graph" class="anchor"></a>5.4. Visualisation of the graph</h2>
<p>The <code><a href="../reference/vis_graph.html">vis_graph()</a></code> function is used for visualization of the graph object. The initial graph object (steady-state) is visualized as following:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">minsamp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="st">   </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb26-3" data-line-number="3">                 <span class="dt">miRNA_count =</span> miRNA_expression,</a>
<a class="sourceLine" id="cb26-4" data-line-number="4">                 <span class="dt">aff_factor =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(energy, seed_type), </a>
<a class="sourceLine" id="cb26-5" data-line-number="5">                 <span class="dt">deg_factor =</span> region) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb26-6" data-line-number="6"><span class="st">   </span><span class="kw"><a href="../reference/vis_graph.html">vis_graph</a></span>(<span class="dt">title =</span> <span class="st">"Minsamp initial Graph"</span>)</a></code></pre></div>
<p><img src="basic_usage_files/figure-html/unnamed-chunk-26-1.png" width="600" style="display: block; margin: auto;"></p>
<p>Also, The graph can be visualized at any step of process, for example, after simulation of 3 cycles the graph will look like:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">minsamp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-2" data-line-number="2"><span class="st">   </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb27-3" data-line-number="3">                 <span class="dt">miRNA_count =</span> miRNA_expression,</a>
<a class="sourceLine" id="cb27-4" data-line-number="4">                 <span class="dt">aff_factor =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(energy, seed_type), </a>
<a class="sourceLine" id="cb27-5" data-line-number="5">                 <span class="dt">deg_factor =</span> region) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-6" data-line-number="6"><span class="st">   </span><span class="kw"><a href="../reference/update_variables.html">update_variables</a></span>(<span class="dt">current_counts =</span> new_counts) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-7" data-line-number="7"><span class="st">   </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dv">3</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-8" data-line-number="8"><span class="st">   </span><span class="kw"><a href="../reference/vis_graph.html">vis_graph</a></span>(<span class="dt">title =</span> <span class="st">"Minsamp Graph After 3 Iteration"</span>)</a></code></pre></div>
<p><img src="basic_usage_files/figure-html/unnamed-chunk-27-1.png" width="600" style="display: block; margin: auto;"></p>
<p>On the other hand, the network of each step can be plotted individually by using <code><a href="../reference/simulate_vis.html">simulate_vis()</a></code> function. <code><a href="../reference/simulate_vis.html">simulate_vis()</a></code> processes the given network just like <code><a href="../reference/simulate.html">simulate()</a></code> function does while saving image of each step.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">minsamp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="st">   </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> Competing_expression, </a>
<a class="sourceLine" id="cb28-3" data-line-number="3">                 <span class="dt">miRNA_count =</span> miRNA_expression,</a>
<a class="sourceLine" id="cb28-4" data-line-number="4">                 <span class="dt">aff_factor =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(energy, seed_type), </a>
<a class="sourceLine" id="cb28-5" data-line-number="5">                 <span class="dt">deg_factor =</span> region) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb28-6" data-line-number="6"><span class="st">   </span><span class="kw"><a href="../reference/update_variables.html">update_variables</a></span>(<span class="dt">current_counts =</span> new_counts) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb28-7" data-line-number="7"><span class="st">   </span><span class="kw"><a href="../reference/simulate_vis.html">simulate_vis</a></span>(<span class="dv">3</span>, <span class="dt">title =</span> <span class="st">"Minsamp Graph After Each Iteration"</span>)</a></code></pre></div>
<div class="figure">
<img src="gifs/small_sample.gif#center" alt="Minsamp with 3 iteration"><p class="caption">Minsamp with 3 iteration</p>
</div>
<p>Note: Animated gif above was generated by online service. Actually, workflow gives the frames which include condition of each iteration. Note that you must use a terminal or online service, if you want to generate the animated gif.</p>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>6. Session Info</h1>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb29-2" data-line-number="2"><span class="co">#&gt; R Under development (unstable) (2019-11-25 r77460)</span></a>
<a class="sourceLine" id="cb29-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb29-4" data-line-number="4"><span class="co">#&gt; Running under: Ubuntu 16.04.6 LTS</span></a>
<a class="sourceLine" id="cb29-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb29-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb29-7" data-line-number="7"><span class="co">#&gt; BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so</span></a>
<a class="sourceLine" id="cb29-8" data-line-number="8"><span class="co">#&gt; LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so</span></a>
<a class="sourceLine" id="cb29-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb29-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb29-11" data-line-number="11"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></a>
<a class="sourceLine" id="cb29-12" data-line-number="12"><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></a>
<a class="sourceLine" id="cb29-13" data-line-number="13"><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></a>
<a class="sourceLine" id="cb29-14" data-line-number="14"><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></a>
<a class="sourceLine" id="cb29-15" data-line-number="15"><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></a>
<a class="sourceLine" id="cb29-16" data-line-number="16"><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb29-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb29-18" data-line-number="18"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb29-19" data-line-number="19"><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></a>
<a class="sourceLine" id="cb29-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb29-21" data-line-number="21"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb29-22" data-line-number="22"><span class="co">#&gt; [1] ceRNAnetsim_0.99.22 tidygraph_1.1.2     dplyr_0.8.3        </span></a>
<a class="sourceLine" id="cb29-23" data-line-number="23"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb29-24" data-line-number="24"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb29-25" data-line-number="25"><span class="co">#&gt;  [1] tidyselect_0.2.5   xfun_0.11          purrr_0.3.3        listenv_0.7.0     </span></a>
<a class="sourceLine" id="cb29-26" data-line-number="26"><span class="co">#&gt;  [5] graphlayouts_0.5.0 colorspace_1.4-1   vctrs_0.2.0        htmltools_0.4.0   </span></a>
<a class="sourceLine" id="cb29-27" data-line-number="27"><span class="co">#&gt;  [9] viridisLite_0.3.0  yaml_2.2.0         utf8_1.1.4         rlang_0.4.2       </span></a>
<a class="sourceLine" id="cb29-28" data-line-number="28"><span class="co">#&gt; [13] pkgdown_1.4.1      pillar_1.4.2       glue_1.3.1         tweenr_1.0.1      </span></a>
<a class="sourceLine" id="cb29-29" data-line-number="29"><span class="co">#&gt; [17] lifecycle_0.1.0    stringr_1.4.0      munsell_0.5.0      gtable_0.3.0      </span></a>
<a class="sourceLine" id="cb29-30" data-line-number="30"><span class="co">#&gt; [21] future_1.15.1      codetools_0.2-16   memoise_1.1.0      evaluate_0.14     </span></a>
<a class="sourceLine" id="cb29-31" data-line-number="31"><span class="co">#&gt; [25] labeling_0.3       knitr_1.26         parallel_4.0.0     fansi_0.4.0       </span></a>
<a class="sourceLine" id="cb29-32" data-line-number="32"><span class="co">#&gt; [29] furrr_0.1.0        Rcpp_1.0.3         backports_1.1.5    scales_1.1.0      </span></a>
<a class="sourceLine" id="cb29-33" data-line-number="33"><span class="co">#&gt; [33] desc_1.2.0         farver_2.0.1       fs_1.3.1           gridExtra_2.3     </span></a>
<a class="sourceLine" id="cb29-34" data-line-number="34"><span class="co">#&gt; [37] ggforce_0.3.1      ggplot2_3.2.1      png_0.1-7          digest_0.6.23     </span></a>
<a class="sourceLine" id="cb29-35" data-line-number="35"><span class="co">#&gt; [41] stringi_1.4.3      ggrepel_0.8.1      polyclip_1.10-0    rprojroot_1.3-2   </span></a>
<a class="sourceLine" id="cb29-36" data-line-number="36"><span class="co">#&gt; [45] grid_4.0.0         cli_1.1.0          tools_4.0.0        magrittr_1.5      </span></a>
<a class="sourceLine" id="cb29-37" data-line-number="37"><span class="co">#&gt; [49] lazyeval_0.2.2     tibble_2.1.3       ggraph_2.0.0       crayon_1.3.4      </span></a>
<a class="sourceLine" id="cb29-38" data-line-number="38"><span class="co">#&gt; [53] tidyr_1.0.0        pkgconfig_2.0.3    zeallot_0.1.0      MASS_7.3-51.4     </span></a>
<a class="sourceLine" id="cb29-39" data-line-number="39"><span class="co">#&gt; [57] assertthat_0.2.1   rmarkdown_1.15     viridis_0.5.1      R6_2.4.1          </span></a>
<a class="sourceLine" id="cb29-40" data-line-number="40"><span class="co">#&gt; [61] globals_0.12.4     igraph_1.2.4.1     compiler_4.0.0</span></a></code></pre></div>
<p>See the other vignettes for more information.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">1. Introduction</a></li>
      <li><a href="#installation">2. Installation</a></li>
      <li><a href="#about-the-data">3. About the data</a></li>
      <li>
<a href="#simulation-via-expression-values-of-mirnas-and-genes-in-minsamp-dataset">4. Simulation via expression values of miRNAs and genes in <code>minsamp</code> dataset</a><ul class="nav nav-pills nav-stacked">
<li><a href="#handle-basic-dataset">4.1. Handle basic dataset</a></li>
      <li><a href="#trigger-a-change">4.2. Trigger a change</a></li>
      <li><a href="#simulate-the-changes-in-graph">4.3. Simulate the changes in graph</a></li>
      <li><a href="#a-special-case-knockdown">4.4. A special case: knockdown</a></li>
      </ul>
</li>
      <li>
<a href="#simulation-via-interaction-factors-in-addition-to-expression-values-of-mirnas-and-genes-in-minsamp-dataset">5. Simulation via interaction factors in addition to expression values of miRNAs and genes in <code>minsamp</code> dataset</a><ul class="nav nav-pills nav-stacked">
<li><a href="#change-expression-level-of-one-or-more-nodes-in-the-graph">5.1. Change expression level of one or more nodes in the graph</a></li>
      <li><a href="#update-the-node-variables-with-edge-variables-">5.2. Update the node variables with edge variables.</a></li>
      <li><a href="#simulate-the-model">5.3. Simulate the model</a></li>
      <li><a href="#visualisation-of-the-graph">5.4. Visualisation of the graph</a></li>
      </ul>
</li>
      <li><a href="#session-info">6. Session Info</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Selcen Ari Yuka, Alper Yilmaz.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'fbd8d79ed49f21fc9ca7b2cf601a3f5b',
    indexName: 'ceRNAnetsim',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
