<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Basic Use of ceRNAnetsim • ceRNAnetsim</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Basic Use of ceRNAnetsim">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-84499524-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-84499524-2');
</script>
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ceRNAnetsim</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-book"></span>
     
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/basic_usage.html">What does ceRNAnetsim do?</a>
    </li>
    <li>
      <a href="../articles/small_sample.html">Small example demonstrating the approach and the method</a>
    </li>
    <li>
      <a href="../articles/convenient_iteration.html">How to calculate proper number of iterations for simulation of the network</a>
    </li>
    <li>
      <a href="../articles/perturbation_sample.html">How to determine most effective node of network in equal conditions</a>
    </li>
    <li>
      <a href="../articles/realexample.html">How does the system do in the presence of interaction factors?</a>
    </li>
    <li>
      <a href="../articles/mirtarbase_example.html">How does the system behave in mirtarbase dataset without interaction factors?</a>
    </li>
    <li>
      <a href="../articles/auxiliary_commands.html">Helper Commands</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/selcenari/ceRNAnetsim">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Basic Use of ceRNAnetsim</h1>
                        <h4 class="author">Selcen Ari</h4>
            
            <h4 class="date">2019-10-17</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/selcenari/ceRNAnetsim/blob/master/vignettes/basic_usage.Rmd"><code>vignettes/basic_usage.Rmd</code></a></small>
      <div class="hidden name"><code>basic_usage.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette demonstrates how to analyse miRNA:Competing interactions via <code>ceRNAnetsim</code> package. The perturbations in the miRNA:target interactions are handled step by step in <code>ceRNAnetsim</code>. The package calculates and simulates regulation of miRNA:competing RNA interactions based on amounts of miRNA and the targets and interaction factors.</p>
<p>The <code>ceRNAnetsim</code> works by executing following steps:</p>
<ul>
<li>The input dataset is prepared for analysis. The first variable is arranged as competing and the second as miRNA.</li>
<li>The dataset is primed with <code><a href="../reference/priming_graph.html">priming_graph()</a></code> function so that it’s converted into a graph. This function makes calculations that are depended on miRNA amount, target (competing) amount and the interaction factors. It determines the efficiency of miRNA to each target and saves that values as edge data. All calculations are performed in edge data. After that, results of calculations are used in node data.</li>
<li>Now, the change in expression level of either miRNA or competing RNA can be introduced with <code><a href="../reference/update_variables.html">update_variables()</a></code> or <code><a href="../reference/update_how.html">update_how()</a></code> functions.</li>
<li>The calculated values in edge are carried to node data through a helper function <code><a href="../reference/update_nodes.html">update_nodes()</a></code>.</li>
<li>The change is perceived as a trigger and network-wise calculations are performed.</li>
<li>The trigger is used for simulation of regulations in <code><a href="../reference/simulate.html">simulate()</a></code> or <code><a href="../reference/simulate_vis.html">simulate_vis()</a></code>.</li>
<li>These processes can generate various outputs such as graphs or graph objects.</li>
</ul>
<p>The workflow of <code>ceRNAnetsim</code> are shown as following:</p>
<p><img src="gifs/model_vignette.png" width="407" style="display: block; margin: auto;"></p>
<p>Firstly the package is loaded:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co">#install.packages("devtools")</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co">#devtools::install_github("selcenari/ceRNAnetsim")</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ceRNAnetsim)</span></code></pre></div>
<div id="about-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#about-the-data" class="anchor"></a>About the data</h2>
<p>Below is the minimal data that can be used with ceRNAnetsim.</p>
<pre><code>#&gt;   competing miRNA Competing_expression miRNA_expression
#&gt; 1     Gene1  Mir1                10000             1000
#&gt; 2     Gene2  Mir1                10000             1000
#&gt; 3     Gene3  Mir1                 5000             1000
#&gt; 4     Gene4  Mir1                10000             1000
#&gt; 5     Gene4  Mir2                10000             2000
#&gt; 6     Gene5  Mir2                 5000             2000
#&gt; 7     Gene6  Mir2                10000             2000</code></pre>
<p>The table is actually constructed by merging three different tables:</p>
<ul>
<li>gene expression data</li>
<li>miRNA expression data</li>
<li>miRNA:gene pairs (might <strong>optionally</strong> contain data about the interaction)</li>
</ul>
<p>So, the <code>basic_data</code> table is constructed by merging following tables:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>gene_expression</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co">#&gt;   competing Competing_expression</span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co">#&gt; 1     Gene1                10000</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co">#&gt; 2     Gene2                10000</span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co">#&gt; 3     Gene3                 5000</span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co">#&gt; 4     Gene4                10000</span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co">#&gt; 5     Gene5                 5000</span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co">#&gt; 6     Gene6                10000</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>mirna_expression</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co">#&gt;   miRNA miRNA_expression</span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co">#&gt; 1  Mir1             1000</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co">#&gt; 2  Mir2             2000</span></span></code></pre></div>
<p>Third table should contain miRNA:gene interactions per row. The <code>ceRNAnetsim</code> will assume first column contains competing RNA names and second column to be miRNA names. If the order is different the user should indicate column names accordingly.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>interaction_simple</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="co">#&gt;   competing miRNA</span></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="co">#&gt; 1     Gene1  Mir1</span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="co">#&gt; 2     Gene2  Mir1</span></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="co">#&gt; 3     Gene3  Mir1</span></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="co">#&gt; 4     Gene4  Mir1</span></span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="co">#&gt; 5     Gene4  Mir2</span></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co">#&gt; 6     Gene5  Mir2</span></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co">#&gt; 7     Gene6  Mir2</span></span></code></pre></div>
<p>The three tables can be joined in R (as shown below) or elsewhere to have interaction and expression data altogether in expected format.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>interaction_simple <span class="op">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="st">  </span><span class="kw">inner_join</span>(gene_expression, <span class="dt">by =</span> <span class="st">"competing"</span>) <span class="op">%&gt;%</span></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="st">  </span><span class="kw">inner_join</span>(mirna_expression, <span class="st">"miRNA"</span>) -&gt;<span class="st"> </span>basic_data</span>
<span id="cb6-4"><a href="#cb6-4"></a></span>
<span id="cb6-5"><a href="#cb6-5"></a>basic_data</span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="co">#&gt;   competing miRNA Competing_expression miRNA_expression</span></span>
<span id="cb6-7"><a href="#cb6-7"></a><span class="co">#&gt; 1     Gene1  Mir1                10000             1000</span></span>
<span id="cb6-8"><a href="#cb6-8"></a><span class="co">#&gt; 2     Gene2  Mir1                10000             1000</span></span>
<span id="cb6-9"><a href="#cb6-9"></a><span class="co">#&gt; 3     Gene3  Mir1                 5000             1000</span></span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="co">#&gt; 4     Gene4  Mir1                10000             1000</span></span>
<span id="cb6-11"><a href="#cb6-11"></a><span class="co">#&gt; 5     Gene4  Mir2                10000             2000</span></span>
<span id="cb6-12"><a href="#cb6-12"></a><span class="co">#&gt; 6     Gene5  Mir2                 5000             2000</span></span>
<span id="cb6-13"><a href="#cb6-13"></a><span class="co">#&gt; 7     Gene6  Mir2                10000             2000</span></span></code></pre></div>
</div>
<div id="what-does-cernanetsim-do" class="section level2">
<h2 class="hasAnchor">
<a href="#what-does-cernanetsim-do" class="anchor"></a>What does ceRNAnetsim do?</h2>
<p>ceRNAnetsim processes your dataset as graph object and simulates competing behaviours of targets when steady-state is perturbed via expression level changes in one or more genes. Let’s go over three steps:</p>
<div id="handle-basic-dataset" class="section level3">
<h3 class="hasAnchor">
<a href="#handle-basic-dataset" class="anchor"></a>1. Handle basic dataset</h3>
<p>In first step, the expression and interaction table is converted into graph/network. <code>tidygraph</code> is used importing the data thus both node and edge data are accessible as tables if needed. <code>priming_graph</code> generates many columns in edge/node table which are mostly for internal use.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co">#Convertion of dataset to graph.</span></span>
<span id="cb7-2"><a href="#cb7-2"></a></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(basic_data, <span class="dt">competing_count =</span> Competing_expression, <span class="dt">miRNA_count =</span>miRNA_expression)</span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt; Warning in priming_graph(basic_data, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&gt; #</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&gt; # A rooted tree</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&gt; #</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co">#&gt;   name  type  node_id initial_count count_pre count_current</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;   &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span id="cb7-12"><a href="#cb7-12"></a><span class="co">#&gt; 1 Gene1 Comp~       1         10000     10000         10000</span></span>
<span id="cb7-13"><a href="#cb7-13"></a><span class="co">#&gt; 2 Gene2 Comp~       2         10000     10000         10000</span></span>
<span id="cb7-14"><a href="#cb7-14"></a><span class="co">#&gt; 3 Gene3 Comp~       3          5000      5000          5000</span></span>
<span id="cb7-15"><a href="#cb7-15"></a><span class="co">#&gt; 4 Gene4 Comp~       4         10000     10000         10000</span></span>
<span id="cb7-16"><a href="#cb7-16"></a><span class="co">#&gt; 5 Gene5 Comp~       5          5000      5000          5000</span></span>
<span id="cb7-17"><a href="#cb7-17"></a><span class="co">#&gt; 6 Gene6 Comp~       6         10000     10000         10000</span></span>
<span id="cb7-18"><a href="#cb7-18"></a><span class="co">#&gt; # ... with 2 more rows, and 1 more variable: changes_variable &lt;chr&gt;</span></span>
<span id="cb7-19"><a href="#cb7-19"></a><span class="co">#&gt; #</span></span>
<span id="cb7-20"><a href="#cb7-20"></a><span class="co">#&gt; # Edge Data: 7 x 19</span></span>
<span id="cb7-21"><a href="#cb7-21"></a><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre~ miRNA_expression</span></span>
<span id="cb7-22"><a href="#cb7-22"></a><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;</span></span>
<span id="cb7-23"><a href="#cb7-23"></a><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000</span></span>
<span id="cb7-24"><a href="#cb7-24"></a><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000</span></span>
<span id="cb7-25"><a href="#cb7-25"></a><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000</span></span>
<span id="cb7-26"><a href="#cb7-26"></a><span class="co">#&gt; # ... with 4 more rows, and 13 more variables: dummy &lt;dbl&gt;,</span></span>
<span id="cb7-27"><a href="#cb7-27"></a><span class="co">#&gt; #   afff_factor &lt;dbl&gt;, degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;,</span></span>
<span id="cb7-28"><a href="#cb7-28"></a><span class="co">#&gt; #   comp_count_pre &lt;dbl&gt;, comp_count_current &lt;dbl&gt;,</span></span>
<span id="cb7-29"><a href="#cb7-29"></a><span class="co">#&gt; #   mirna_count_list &lt;list&gt;, mirna_count_pre &lt;dbl&gt;,</span></span>
<span id="cb7-30"><a href="#cb7-30"></a><span class="co">#&gt; #   mirna_count_current &lt;dbl&gt;, mirna_count_per_dep &lt;dbl&gt;,</span></span>
<span id="cb7-31"><a href="#cb7-31"></a><span class="co">#&gt; #   effect_current &lt;dbl&gt;, effect_pre &lt;dbl&gt;, effect_list &lt;list&gt;</span></span></code></pre></div>
</div>
<div id="trigger-a-change" class="section level3">
<h3 class="hasAnchor">
<a href="#trigger-a-change" class="anchor"></a>2. Trigger a change</h3>
<p><code>update_how</code> function can be used to simulate a change in the network. (If multiple chnages are aimed to be used as trigger, <code><a href="../reference/update_variables.html">update_variables()</a></code> function should be used).</p>
<p>In the example below, expression level of “Gene2” is increased to two-fold.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(basic_data, <span class="dt">competing_count =</span> Competing_expression, </span>
<span id="cb8-3"><a href="#cb8-3"></a>              <span class="dt">miRNA_count =</span>miRNA_expression) <span class="op">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"Gene2"</span>, <span class="dt">how=</span><span class="dv">2</span>)</span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="co">#&gt; Warning in priming_graph(basic_data, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></span>
<span id="cb8-6"><a href="#cb8-6"></a><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="co">#&gt; #</span></span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="co">#&gt; # A rooted tree</span></span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="co">#&gt; #</span></span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="co">#&gt;   name  type  node_id initial_count count_pre count_current</span></span>
<span id="cb8-12"><a href="#cb8-12"></a><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;   &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span id="cb8-13"><a href="#cb8-13"></a><span class="co">#&gt; 1 Gene1 Comp~       1         10000     10000         10000</span></span>
<span id="cb8-14"><a href="#cb8-14"></a><span class="co">#&gt; 2 Gene2 Comp~       2         10000     10000         20000</span></span>
<span id="cb8-15"><a href="#cb8-15"></a><span class="co">#&gt; 3 Gene3 Comp~       3          5000      5000          5000</span></span>
<span id="cb8-16"><a href="#cb8-16"></a><span class="co">#&gt; 4 Gene4 Comp~       4         10000     10000         10000</span></span>
<span id="cb8-17"><a href="#cb8-17"></a><span class="co">#&gt; 5 Gene5 Comp~       5          5000      5000          5000</span></span>
<span id="cb8-18"><a href="#cb8-18"></a><span class="co">#&gt; 6 Gene6 Comp~       6         10000     10000         10000</span></span>
<span id="cb8-19"><a href="#cb8-19"></a><span class="co">#&gt; # ... with 2 more rows, and 1 more variable: changes_variable &lt;chr&gt;</span></span>
<span id="cb8-20"><a href="#cb8-20"></a><span class="co">#&gt; #</span></span>
<span id="cb8-21"><a href="#cb8-21"></a><span class="co">#&gt; # Edge Data: 7 x 19</span></span>
<span id="cb8-22"><a href="#cb8-22"></a><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre~ miRNA_expression</span></span>
<span id="cb8-23"><a href="#cb8-23"></a><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;</span></span>
<span id="cb8-24"><a href="#cb8-24"></a><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000</span></span>
<span id="cb8-25"><a href="#cb8-25"></a><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000</span></span>
<span id="cb8-26"><a href="#cb8-26"></a><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000</span></span>
<span id="cb8-27"><a href="#cb8-27"></a><span class="co">#&gt; # ... with 4 more rows, and 13 more variables: dummy &lt;dbl&gt;,</span></span>
<span id="cb8-28"><a href="#cb8-28"></a><span class="co">#&gt; #   afff_factor &lt;dbl&gt;, degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;,</span></span>
<span id="cb8-29"><a href="#cb8-29"></a><span class="co">#&gt; #   comp_count_pre &lt;dbl&gt;, comp_count_current &lt;dbl&gt;,</span></span>
<span id="cb8-30"><a href="#cb8-30"></a><span class="co">#&gt; #   mirna_count_list &lt;list&gt;, mirna_count_pre &lt;dbl&gt;,</span></span>
<span id="cb8-31"><a href="#cb8-31"></a><span class="co">#&gt; #   mirna_count_current &lt;dbl&gt;, mirna_count_per_dep &lt;dbl&gt;,</span></span>
<span id="cb8-32"><a href="#cb8-32"></a><span class="co">#&gt; #   effect_current &lt;dbl&gt;, effect_pre &lt;dbl&gt;, effect_list &lt;list&gt;</span></span></code></pre></div>
<p>You can see the current count of Gene2 node is 20000 and its change is denoted as “Up” in <code>changes_variable</code> column in node table data.</p>
</div>
<div id="simulate-the-changes-in-graph" class="section level3">
<h3 class="hasAnchor">
<a href="#simulate-the-changes-in-graph" class="anchor"></a>3. Simulate the changes in graph</h3>
<p>Finally, with the help of <code>simulate</code> function, the effect of expression change (<em>i.e.</em> the trigger) on overall network. The example code advances only for 5 cycles.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a></span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(basic_data, <span class="dt">competing_count =</span> Competing_expression, </span>
<span id="cb9-3"><a href="#cb9-3"></a>              <span class="dt">miRNA_count =</span>miRNA_expression) <span class="op">%&gt;%</span></span>
<span id="cb9-4"><a href="#cb9-4"></a><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"Gene2"</span>, <span class="dt">how=</span><span class="dv">2</span>) <span class="op">%&gt;%</span></span>
<span id="cb9-5"><a href="#cb9-5"></a><span class="st">  </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dt">cycle =</span> <span class="dv">5</span>)</span>
<span id="cb9-6"><a href="#cb9-6"></a><span class="co">#&gt; Warning in priming_graph(basic_data, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></span>
<span id="cb9-7"><a href="#cb9-7"></a><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></span>
<span id="cb9-8"><a href="#cb9-8"></a><span class="co">#&gt; #</span></span>
<span id="cb9-9"><a href="#cb9-9"></a><span class="co">#&gt; # A rooted tree</span></span>
<span id="cb9-10"><a href="#cb9-10"></a><span class="co">#&gt; #</span></span>
<span id="cb9-11"><a href="#cb9-11"></a><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></span>
<span id="cb9-12"><a href="#cb9-12"></a><span class="co">#&gt;   name  type  node_id initial_count count_pre count_current</span></span>
<span id="cb9-13"><a href="#cb9-13"></a><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;   &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span id="cb9-14"><a href="#cb9-14"></a><span class="co">#&gt; 1 Gene1 Comp~       1         10000    10062.        10062.</span></span>
<span id="cb9-15"><a href="#cb9-15"></a><span class="co">#&gt; 2 Gene2 Comp~       2         10000    19845.        19845.</span></span>
<span id="cb9-16"><a href="#cb9-16"></a><span class="co">#&gt; 3 Gene3 Comp~       3          5000     5031.         5031.</span></span>
<span id="cb9-17"><a href="#cb9-17"></a><span class="co">#&gt; 4 Gene4 Comp~       4         10000    10059.        10059.</span></span>
<span id="cb9-18"><a href="#cb9-18"></a><span class="co">#&gt; 5 Gene5 Comp~       5          5000     5001.         5001.</span></span>
<span id="cb9-19"><a href="#cb9-19"></a><span class="co">#&gt; 6 Gene6 Comp~       6         10000    10002.        10002.</span></span>
<span id="cb9-20"><a href="#cb9-20"></a><span class="co">#&gt; # ... with 2 more rows, and 1 more variable: changes_variable &lt;chr&gt;</span></span>
<span id="cb9-21"><a href="#cb9-21"></a><span class="co">#&gt; #</span></span>
<span id="cb9-22"><a href="#cb9-22"></a><span class="co">#&gt; # Edge Data: 7 x 20</span></span>
<span id="cb9-23"><a href="#cb9-23"></a><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre~ miRNA_expression</span></span>
<span id="cb9-24"><a href="#cb9-24"></a><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;</span></span>
<span id="cb9-25"><a href="#cb9-25"></a><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000</span></span>
<span id="cb9-26"><a href="#cb9-26"></a><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000</span></span>
<span id="cb9-27"><a href="#cb9-27"></a><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000</span></span>
<span id="cb9-28"><a href="#cb9-28"></a><span class="co">#&gt; # ... with 4 more rows, and 14 more variables: dummy &lt;dbl&gt;,</span></span>
<span id="cb9-29"><a href="#cb9-29"></a><span class="co">#&gt; #   afff_factor &lt;dbl&gt;, degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;,</span></span>
<span id="cb9-30"><a href="#cb9-30"></a><span class="co">#&gt; #   comp_count_pre &lt;dbl&gt;, comp_count_current &lt;dbl&gt;,</span></span>
<span id="cb9-31"><a href="#cb9-31"></a><span class="co">#&gt; #   mirna_count_list &lt;list&gt;, mirna_count_pre &lt;dbl&gt;,</span></span>
<span id="cb9-32"><a href="#cb9-32"></a><span class="co">#&gt; #   mirna_count_current &lt;dbl&gt;, mirna_count_per_dep &lt;dbl&gt;,</span></span>
<span id="cb9-33"><a href="#cb9-33"></a><span class="co">#&gt; #   effect_current &lt;dbl&gt;, effect_pre &lt;dbl&gt;, effect_list &lt;list&gt;,</span></span>
<span id="cb9-34"><a href="#cb9-34"></a><span class="co">#&gt; #   mirna_count_per_comp &lt;dbl&gt;</span></span></code></pre></div>
<p><code>count_current</code> column indicate results after 5 cycle of calculations for each node. You can see the gene expression changes after this perturbation and simulation. This table can be obtained easily at following:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(basic_data, <span class="dt">competing_count =</span> Competing_expression, </span>
<span id="cb10-2"><a href="#cb10-2"></a>              <span class="dt">miRNA_count =</span>miRNA_expression) <span class="op">%&gt;%</span></span>
<span id="cb10-3"><a href="#cb10-3"></a><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"Gene2"</span>, <span class="dt">how=</span><span class="dv">2</span>) <span class="op">%&gt;%</span></span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="st">  </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dt">cycle =</span> <span class="dv">5</span>)<span class="op">%&gt;%</span></span>
<span id="cb10-5"><a href="#cb10-5"></a><span class="st">  </span><span class="kw">as_tibble</span>()<span class="op">%&gt;%</span></span>
<span id="cb10-6"><a href="#cb10-6"></a><span class="st">  </span><span class="kw">select</span>(name, initial_count, count_current)</span>
<span id="cb10-7"><a href="#cb10-7"></a><span class="co">#&gt; Warning in priming_graph(basic_data, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></span>
<span id="cb10-8"><a href="#cb10-8"></a><span class="co">#&gt; # A tibble: 8 x 3</span></span>
<span id="cb10-9"><a href="#cb10-9"></a><span class="co">#&gt;   name  initial_count count_current</span></span>
<span id="cb10-10"><a href="#cb10-10"></a><span class="co">#&gt;   &lt;chr&gt;         &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span id="cb10-11"><a href="#cb10-11"></a><span class="co">#&gt; 1 Gene1         10000        10062.</span></span>
<span id="cb10-12"><a href="#cb10-12"></a><span class="co">#&gt; 2 Gene2         10000        19845.</span></span>
<span id="cb10-13"><a href="#cb10-13"></a><span class="co">#&gt; 3 Gene3          5000         5031.</span></span>
<span id="cb10-14"><a href="#cb10-14"></a><span class="co">#&gt; 4 Gene4         10000        10059.</span></span>
<span id="cb10-15"><a href="#cb10-15"></a><span class="co">#&gt; 5 Gene5          5000         5001.</span></span>
<span id="cb10-16"><a href="#cb10-16"></a><span class="co">#&gt; 6 Gene6         10000        10002.</span></span>
<span id="cb10-17"><a href="#cb10-17"></a><span class="co">#&gt; 7 Mir1           1000         1000 </span></span>
<span id="cb10-18"><a href="#cb10-18"></a><span class="co">#&gt; 8 Mir2           2000         2000</span></span></code></pre></div>
</div>
</div>
<div id="a-special-case-knockdown" class="section level2">
<h2 class="hasAnchor">
<a href="#a-special-case-knockdown" class="anchor"></a>A special case: knockdown</h2>
<p>ceRNAnetsim also provides the simulation of gene knockdown in the network. In normal conditions, when a gene is up or down regulated, it is considered that amounts of gene transcripts change depended on interactions. But, the transcripts of the gene are not observed in the system when it is knocked down. To achieve this case, you just need to define <code>how</code> argument to 0 (zero) in <code>update_how</code> function.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(basic_data, <span class="dt">competing_count =</span> Competing_expression, </span>
<span id="cb11-3"><a href="#cb11-3"></a>              <span class="dt">miRNA_count =</span>miRNA_expression) <span class="op">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"Gene2"</span>, <span class="dt">how=</span><span class="dv">0</span>) <span class="op">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5"></a><span class="st">  </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dt">cycle =</span> <span class="dv">5</span>)</span>
<span id="cb11-6"><a href="#cb11-6"></a><span class="co">#&gt; Warning in priming_graph(basic_data, competing_count = Competing_expression, : First column is processed as competing and the second as miRNA.</span></span>
<span id="cb11-7"><a href="#cb11-7"></a><span class="co">#&gt; # A tbl_graph: 8 nodes and 7 edges</span></span>
<span id="cb11-8"><a href="#cb11-8"></a><span class="co">#&gt; #</span></span>
<span id="cb11-9"><a href="#cb11-9"></a><span class="co">#&gt; # A rooted tree</span></span>
<span id="cb11-10"><a href="#cb11-10"></a><span class="co">#&gt; #</span></span>
<span id="cb11-11"><a href="#cb11-11"></a><span class="co">#&gt; # Node Data: 8 x 7 (active)</span></span>
<span id="cb11-12"><a href="#cb11-12"></a><span class="co">#&gt;   name  type  node_id initial_count count_pre count_current</span></span>
<span id="cb11-13"><a href="#cb11-13"></a><span class="co">#&gt;   &lt;chr&gt; &lt;chr&gt;   &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span id="cb11-14"><a href="#cb11-14"></a><span class="co">#&gt; 1 Gene1 Comp~       1         10000     9886.         9886.</span></span>
<span id="cb11-15"><a href="#cb11-15"></a><span class="co">#&gt; 2 Gene2 Comp~       2         10000        0             0 </span></span>
<span id="cb11-16"><a href="#cb11-16"></a><span class="co">#&gt; 3 Gene3 Comp~       3          5000     4943.         4943.</span></span>
<span id="cb11-17"><a href="#cb11-17"></a><span class="co">#&gt; 4 Gene4 Comp~       4         10000     9891.         9891.</span></span>
<span id="cb11-18"><a href="#cb11-18"></a><span class="co">#&gt; 5 Gene5 Comp~       5          5000     4998.         4998.</span></span>
<span id="cb11-19"><a href="#cb11-19"></a><span class="co">#&gt; 6 Gene6 Comp~       6         10000     9997.         9997.</span></span>
<span id="cb11-20"><a href="#cb11-20"></a><span class="co">#&gt; # ... with 2 more rows, and 1 more variable: changes_variable &lt;chr&gt;</span></span>
<span id="cb11-21"><a href="#cb11-21"></a><span class="co">#&gt; #</span></span>
<span id="cb11-22"><a href="#cb11-22"></a><span class="co">#&gt; # Edge Data: 7 x 20</span></span>
<span id="cb11-23"><a href="#cb11-23"></a><span class="co">#&gt;    from    to Competing_name miRNA_name Competing_expre~ miRNA_expression</span></span>
<span id="cb11-24"><a href="#cb11-24"></a><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;</span></span>
<span id="cb11-25"><a href="#cb11-25"></a><span class="co">#&gt; 1     1     7 Gene1          Mir1                  10000             1000</span></span>
<span id="cb11-26"><a href="#cb11-26"></a><span class="co">#&gt; 2     2     7 Gene2          Mir1                  10000             1000</span></span>
<span id="cb11-27"><a href="#cb11-27"></a><span class="co">#&gt; 3     3     7 Gene3          Mir1                   5000             1000</span></span>
<span id="cb11-28"><a href="#cb11-28"></a><span class="co">#&gt; # ... with 4 more rows, and 14 more variables: dummy &lt;dbl&gt;,</span></span>
<span id="cb11-29"><a href="#cb11-29"></a><span class="co">#&gt; #   afff_factor &lt;dbl&gt;, degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;,</span></span>
<span id="cb11-30"><a href="#cb11-30"></a><span class="co">#&gt; #   comp_count_pre &lt;dbl&gt;, comp_count_current &lt;dbl&gt;,</span></span>
<span id="cb11-31"><a href="#cb11-31"></a><span class="co">#&gt; #   mirna_count_list &lt;list&gt;, mirna_count_pre &lt;dbl&gt;,</span></span>
<span id="cb11-32"><a href="#cb11-32"></a><span class="co">#&gt; #   mirna_count_current &lt;dbl&gt;, mirna_count_per_dep &lt;dbl&gt;,</span></span>
<span id="cb11-33"><a href="#cb11-33"></a><span class="co">#&gt; #   effect_current &lt;dbl&gt;, effect_pre &lt;dbl&gt;, effect_list &lt;list&gt;,</span></span>
<span id="cb11-34"><a href="#cb11-34"></a><span class="co">#&gt; #   mirna_count_per_comp &lt;dbl&gt;</span></span></code></pre></div>
<p>So, if Gene2 is knocked down, there will be more miRNA (Mir1 to be exact) available for Gene1, Gene3 and Gene4, thus lowering their transcript levels. Since Gene4 is has lower expression level, we can observe minute changes in Gene5 and Gene6 levels due to more miRNA (Mir2) being available for them. These changes can be observed in <code>current_count</code> column.</p>
<p>Briefly, ceRNAnetsim utilizes the change(s) as trigger and calculates regulation of targets according to miRNA:target and target:total target ratios.</p>
<p>See the other vignettes for more information.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#about-the-data">About the data</a></li>
      <li><a href="#what-does-cernanetsim-do">What does ceRNAnetsim do?</a></li>
      <li><a href="#a-special-case-knockdown">A special case: knockdown</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Selcen Ari Yuka, Alper Yilmaz.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'fbd8d79ed49f21fc9ca7b2cf601a3f5b',
    indexName: 'ceRNAnetsim',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
