<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A TCGA dataset application • ceRNAnetsim</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="A TCGA dataset application">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-84499524-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-84499524-2');
</script>
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ceRNAnetsim</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa fas fa-book"></span>
     
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/basic_usage.html">What does ceRNAnetsim do?</a>
    </li>
    <li>
      <a href="../articles/convenient_iteration.html">How to calculate proper number of iterations for simulation of the network</a>
    </li>
    <li>
      <a href="../articles/mirtarbase_example.html">How does the system behave in mirtarbase dataset without interaction factors?</a>
    </li>
    <li>
      <a href="../articles/auxiliary_commands.html">Helper Commands</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper-o"></span>
     
    news
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    functions
  </a>
</li>
<li>
  <a href="https://github.com/selcenari/ceRNAnetsim">
    <span class="fa fa-github fa-lg"></span>
     
    github
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>A TCGA dataset application</h1>
                        <h4 class="author">Selcen Ari</h4>
            
            <h4 class="date">2019-11-25</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/selcenari/ceRNAnetsim/blob/master/vignettes/mirtarbase_example.Rmd"><code>vignettes/mirtarbase_example.Rmd</code></a></small>
      <div class="hidden name"><code>mirtarbase_example.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>1. Introduction</h1>
<p>This vignette is about the integration of gene and miRNA pairs and their expression dataset and analysis. The sample dataset in this demonstration, which contains human miRNA:target pairs, was retrieved from <a href="http://mirtarbase.mbc.nctu.edu.tw/php/download.php">miRTarBase</a> website (Release 7.0).</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ceRNAnetsim)</a></code></pre></div>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>2. Installation</h1>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">    <span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">"ceRNAnetsim"</span>)</a></code></pre></div>
</div>
<div id="integration-of-dataset-which-includes-only-mirna-and-gene-expression-values" class="section level1">
<h1 class="hasAnchor">
<a href="#integration-of-dataset-which-includes-only-mirna-and-gene-expression-values" class="anchor"></a>3. Integration of dataset which includes only miRNA and gene expression values</h1>
<div id="mirnatarget-pairs" class="section level2">
<h2 class="hasAnchor">
<a href="#mirnatarget-pairs" class="anchor"></a>3.1. miRNA:target pairs</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"mirtarbasegene"</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(mirtarbasegene)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; # A tibble: 6 x 2</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt;   miRNA           Target </span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt;   &lt;chr&gt;           &lt;chr&gt;  </span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt; 1 hsa-miR-20a-5p  HIF1A  </span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt; 2 hsa-miR-146a-5p CXCR4  </span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt; 3 hsa-miR-122-5p  CYP7A1 </span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt; 4 hsa-miR-222-3p  STAT5A </span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; 5 hsa-miR-21-5p   RASGRP1</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt; 6 hsa-miR-148a-3p DNMT1</span></a></code></pre></div>
<p><strong>NOTE</strong> if the mirna:target dataset includes miRNA genes as targets, the <code><a href="../reference/priming_graph.html">priming_graph()</a></code> function can fail. Because, the function define to miRNAs and targets without distinguishing between uppercase or lowercase.</p>
</div>
<div id="gene-expression-in-normal-and-tumor-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-expression-in-normal-and-tumor-samples" class="anchor"></a>3.2. Gene expression in normal and tumor samples</h2>
<p>The gene and mirna expression counts of patient barcoded with TCGA-E9-A1N5 is retrieved from TCGA via <code>TCGAbiolinks</code> package <span class="citation">(Colaprico et al. 2015)</span> from <code>Bioconductor</code>. The instructions of retrieving data can be found at <code>TCGAbiolinks</code> manual.</p>
<p>For this step you don’t have to use TCGA data, any other source or package can be utilized.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"TCGA_E9_A1N5_normal"</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(TCGA_E9_A1N5_normal)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; # A tibble: 6 x 7</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt;   patient sample barcode definition ensembl_gene_id external_gene_n…</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt;           &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt; 1 TCGA-E… TCGA-… TCGA-E… Solid Tis… ENSG00000000003 TSPAN6          </span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt; 2 TCGA-E… TCGA-… TCGA-E… Solid Tis… ENSG00000000005 TNMD            </span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; 3 TCGA-E… TCGA-… TCGA-E… Solid Tis… ENSG00000000419 DPM1            </span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; 4 TCGA-E… TCGA-… TCGA-E… Solid Tis… ENSG00000000457 SCYL3           </span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt; 5 TCGA-E… TCGA-… TCGA-E… Solid Tis… ENSG00000000460 C1orf112        </span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt; 6 TCGA-E… TCGA-… TCGA-E… Solid Tis… ENSG00000000938 FGR             </span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; # … with 1 more variable: gene_expression &lt;dbl&gt;</span></a></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"TCGA_E9_A1N5_tumor"</span>)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(TCGA_E9_A1N5_tumor)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; # A tibble: 6 x 7</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt;   patient sample barcode definition ensembl_gene_id external_gene_n…</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt;           &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt; 1 TCGA-E… TCGA-… TCGA-E… Primary s… ENSG00000000003 TSPAN6          </span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt; 2 TCGA-E… TCGA-… TCGA-E… Primary s… ENSG00000000005 TNMD            </span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; 3 TCGA-E… TCGA-… TCGA-E… Primary s… ENSG00000000419 DPM1            </span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt; 4 TCGA-E… TCGA-… TCGA-E… Primary s… ENSG00000000457 SCYL3           </span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt; 5 TCGA-E… TCGA-… TCGA-E… Primary s… ENSG00000000460 C1orf112        </span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt; 6 TCGA-E… TCGA-… TCGA-E… Primary s… ENSG00000000938 FGR             </span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt; # … with 1 more variable: gene_expression &lt;dbl&gt;</span></a></code></pre></div>
</div>
<div id="mirna-expression-data" class="section level2">
<h2 class="hasAnchor">
<a href="#mirna-expression-data" class="anchor"></a>3.3. miRNA expression data</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"TCGA_E9_A1N5_mirnatumor"</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(TCGA_E9_A1N5_mirnatumor)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt; # A tibble: 6 x 6</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt;   barcode                mirbase_ID   miRNA        Precusor total_read total_RPM</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt;   &lt;chr&gt;                  &lt;chr&gt;        &lt;chr&gt;        &lt;chr&gt;         &lt;int&gt;     &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; 1 TCGA-E9-A1N5-01A-11R-… MIMAT0000062 hsa-let-7a-… MI00000…      45725  20802.  </span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; 2 TCGA-E9-A1N5-01A-11R-… MIMAT0004481 hsa-let-7a-… MI00000…        100     45.5 </span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; 3 TCGA-E9-A1N5-01A-11R-… MIMAT0010195 hsa-let-7a-… MI00000…          6      2.73</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; 4 TCGA-E9-A1N5-01A-11R-… MIMAT0000063 hsa-let-7b-… MI00000…      43489  19785.  </span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; 5 TCGA-E9-A1N5-01A-11R-… MIMAT0004482 hsa-let-7b-… MI00000…        126     57.3 </span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt; 6 TCGA-E9-A1N5-01A-11R-… MIMAT0000064 hsa-let-7c-… MI00000…       2002    911.</span></a></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"TCGA_E9_A1N5_mirnanormal"</span>)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(TCGA_E9_A1N5_mirnanormal)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; # A tibble: 6 x 6</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt;   barcode                mirbase_ID   miRNA        Precusor total_read total_RPM</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt;   &lt;chr&gt;                  &lt;chr&gt;        &lt;chr&gt;        &lt;chr&gt;         &lt;int&gt;     &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; 1 TCGA-E9-A1N5-11A-41R-… MIMAT0000062 hsa-let-7a-… MI00000…      67599   37068. </span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; 2 TCGA-E9-A1N5-11A-41R-… MIMAT0004481 hsa-let-7a-… MI00000…        132      72.4</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt; 3 TCGA-E9-A1N5-11A-41R-… MIMAT0010195 hsa-let-7a-… MI00000…         57      31.3</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt; 4 TCGA-E9-A1N5-11A-41R-… MIMAT0000063 hsa-let-7b-… MI00000…      47266   25918. </span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt; 5 TCGA-E9-A1N5-11A-41R-… MIMAT0004482 hsa-let-7b-… MI00000…        126      69.1</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt; 6 TCGA-E9-A1N5-11A-41R-… MIMAT0000064 hsa-let-7c-… MI00000…      14554    7981.</span></a></code></pre></div>
<p>Here’s the summary of size of each dataset</p>
<table class="table">
<thead><tr class="header">
<th>Dataset name</th>
<th>Number of rows</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>mirtarbasegene</code></td>
<td>380627</td>
</tr>
<tr class="even">
<td><code>TCGA_E9_A1N5_normal</code></td>
<td>56830</td>
</tr>
<tr class="odd">
<td><code>TCGA_E9_A1N5_tumor</code></td>
<td>56830</td>
</tr>
<tr class="even">
<td><code>TCGA_E9_A1N5_mirnanormal</code></td>
<td>750</td>
</tr>
<tr class="odd">
<td><code>TCGA_E9_A1N5_mirnatumor</code></td>
<td>648</td>
</tr>
</tbody>
</table>
</div>
<div id="integrating-and-analysing-data" class="section level2">
<h2 class="hasAnchor">
<a href="#integrating-and-analysing-data" class="anchor"></a>3.4. Integrating and analysing data</h2>
<p>All of these datasets are integrated using the code below resulting in miRNA:target dataset that contains miRNA and gene expression values.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">TCGA_E9_A1N5_mirnanormal <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="st">  </span><span class="kw">inner_join</span>(mirtarbasegene, <span class="dt">by=</span> <span class="st">"miRNA"</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="st">  </span><span class="kw">inner_join</span>(TCGA_E9_A1N5_normal, </a>
<a class="sourceLine" id="cb8-4" data-line-number="4">             <span class="dt">by =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Target"</span>=<span class="st"> "external_gene_name"</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="st">  </span><span class="kw">select</span>(Target, miRNA, total_read, gene_expression) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="st">  </span><span class="kw">distinct</span>() -&gt;<span class="st"> </span>TCGA_E9_A1N5_mirnagene</a></code></pre></div>
<p>Note: Some of genes have expression values more than one because some of tissue samples were sequenced in two medium separately. So, we select maximum expression values of that genes at following:</p>
<pre><code>#&gt; # A tibble: 26 x 3
#&gt; # Groups:   Target, miRNA [26]
#&gt;    Target  miRNA               n
#&gt;    &lt;chr&gt;   &lt;chr&gt;           &lt;int&gt;
#&gt;  1 COG8    hsa-miR-186-5p      2
#&gt;  2 GOLGA8M hsa-miR-1270        2
#&gt;  3 GOLGA8M hsa-miR-5703        2
#&gt;  4 MATR3   hsa-let-7e-5p       2
#&gt;  5 MATR3   hsa-miR-1-3p        2
#&gt;  6 MATR3   hsa-miR-10b-3p      2
#&gt;  7 MATR3   hsa-miR-125b-5p     2
#&gt;  8 MATR3   hsa-miR-149-5p      2
#&gt;  9 MATR3   hsa-miR-155-5p      2
#&gt; 10 MATR3   hsa-miR-16-1-3p     2
#&gt; # … with 16 more rows</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(TCGA_E9_A1N5_mirnagene)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co">#&gt; # A tibble: 6 x 4</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">#&gt;   Target  miRNA         total_read gene_expression</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt;   &lt;chr&gt;   &lt;chr&gt;              &lt;int&gt;           &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#&gt; 1 CDK6    hsa-let-7a-5p      67599            4669</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#&gt; 2 MYC     hsa-let-7a-5p      67599           11593</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt; 3 BCL2    hsa-let-7a-5p      67599            2445</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt; 4 NKIRAS2 hsa-let-7a-5p      67599            1519</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="co">#&gt; 5 ITGB3   hsa-let-7a-5p      67599             196</span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="co">#&gt; 6 NF2     hsa-let-7a-5p      67599            1755</span></a></code></pre></div>
<p>When we compared the two gene expression dataset of TCGA-E9A1N5 patient, and selected a gene which has 30-fold increased expression, (gene name: HIST1H3H), this gene node will be used in the example. Note that the selected node must not be isolated one. If the an isolated node is selected the change in expression will not propagate in network. (You can see commands for node selection in the vignette <a href="https://selcenari.github.io/ceRNAnetsim/articles/auxiliary_commands.html">The auxiliary commands which can help to the users</a>)</p>
<p>Optionally, you can filter the low expressed gene nodes because they are not effective elements.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">TCGA_E9_A1N5_mirnagene<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(gene_expression <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>)-&gt;TCGA_E9_A1N5_mirnagene</a></code></pre></div>
<p>The analysis is performed based on amounts of miRNAs and targets as seen. Firstly, we tried to find optimal iteration for the network when simulation start with <em>HIST1H3H</em> node.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"></a>
<a class="sourceLine" id="cb12-2" data-line-number="2">TCGA_E9_A1N5_mirnagene <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> gene_expression, </a>
<a class="sourceLine" id="cb12-4" data-line-number="4">                <span class="dt">miRNA_count =</span> total_read) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"HIST1H3H"</span>, <span class="dt">how =</span><span class="dv">30</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dv">10</span>) -&gt;simulation_res_HIST</a>
<a class="sourceLine" id="cb12-7" data-line-number="7"></a>
<a class="sourceLine" id="cb12-8" data-line-number="8">simulation_res_HIST<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="st">  </span><span class="kw"><a href="../reference/find_iteration.html">find_iteration</a></span>(<span class="dt">plot=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="mirtarbase_example_files/figure-html/unnamed-chunk-12-1.png" width="576" style="display: block; margin: auto;"></p>
<p>The graph was shown that the change in expression level of HIST1H3H results in weak perturbation efficiency, despite 30-fold change. The code shown below can be used for calculation of fold changes after simulation HIST1H3H gene to 30 fold:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"></a>
<a class="sourceLine" id="cb13-2" data-line-number="2">simulation_res_HIST<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="st">  </span><span class="kw">as_tibble</span>()<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">FC=</span> count_current<span class="op">/</span>initial_count)<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="st">  </span><span class="kw">arrange</span>(<span class="kw">desc</span>(FC))</a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="co">#&gt; # A tibble: 13,432 x 8</span></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="co">#&gt;    name  type  node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;   &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="co">#&gt;  1 HIST… Comp…    9705            27     808.          808.  Up              </span></a>
<a class="sourceLine" id="cb13-10" data-line-number="10"><span class="co">#&gt;  2 KLHL… Comp…     618           579     579.          579.  Up              </span></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="co">#&gt;  3 YWHAZ Comp…     393         15845   15845.        15845.  Down            </span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12"><span class="co">#&gt;  4 SOCS1 Comp…     433            27      27.0          27.0 Up              </span></a>
<a class="sourceLine" id="cb13-13" data-line-number="13"><span class="co">#&gt;  5 NNT   Comp…    3404          1909    1909.         1909.  Up              </span></a>
<a class="sourceLine" id="cb13-14" data-line-number="14"><span class="co">#&gt;  6 PPIP… Comp…    7084          2342    2342.         2342.  Up              </span></a>
<a class="sourceLine" id="cb13-15" data-line-number="15"><span class="co">#&gt;  7 LYVE1 Comp…   11198           916     916.          916.  Up              </span></a>
<a class="sourceLine" id="cb13-16" data-line-number="16"><span class="co">#&gt;  8 IL10  Comp…    1639            13      13.0          13.0 Up              </span></a>
<a class="sourceLine" id="cb13-17" data-line-number="17"><span class="co">#&gt;  9 L1CAM Comp…    2651            60      60.0          60.0 Down            </span></a>
<a class="sourceLine" id="cb13-18" data-line-number="18"><span class="co">#&gt; 10 IFIT… Comp…    6030          1550    1550.         1550.  Up              </span></a>
<a class="sourceLine" id="cb13-19" data-line-number="19"><span class="co">#&gt; # … with 13,422 more rows, and 1 more variable: FC &lt;dbl&gt;</span></a></code></pre></div>
<p>And then, we tried to simulate the network with the gene which has higher expression value. For this, we selected <em>ACTB</em> node as shown in <a href="https://selcenari.github.io/ceRNAnetsim/articles/auxiliary_commands.html">The auxiliary commands which can help to the users</a></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">simulation_res_ACTB &lt;-<span class="st"> </span>TCGA_E9_A1N5_mirnagene <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> gene_expression, </a>
<a class="sourceLine" id="cb14-3" data-line-number="3">                <span class="dt">miRNA_count =</span> total_read) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="dt">node_name =</span> <span class="st">"ACTB"</span>, <span class="dt">how =</span><span class="fl">1.87</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="../reference/simulate.html">simulate</a></span>(<span class="dv">10</span>)</a>
<a class="sourceLine" id="cb14-6" data-line-number="6"></a>
<a class="sourceLine" id="cb14-7" data-line-number="7">simulation_res_ACTB<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="st">  </span><span class="kw"><a href="../reference/find_iteration.html">find_iteration</a></span>(<span class="dt">plot=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="mirtarbase_example_files/figure-html/unnamed-chunk-14-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Following codes are shown entire gene fold changes after simulation ACTB gene to 1.87 fold:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"></a>
<a class="sourceLine" id="cb15-2" data-line-number="2">simulation_res_ACTB<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="st">  </span><span class="kw">as_tibble</span>()<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">FC=</span> count_current<span class="op">/</span>initial_count)<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="st">  </span><span class="kw">arrange</span>(<span class="kw">desc</span>(FC))</a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="co">#&gt; # A tibble: 13,432 x 8</span></a>
<a class="sourceLine" id="cb15-7" data-line-number="7"><span class="co">#&gt;    name  type  node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb15-8" data-line-number="8"><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;   &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb15-9" data-line-number="9"><span class="co">#&gt;  1 ACTB  Comp…      84        101917  183710.       183710.  Up              </span></a>
<a class="sourceLine" id="cb15-10" data-line-number="10"><span class="co">#&gt;  2 ADIP… Comp…     319          3015    3026.         3026.  Up              </span></a>
<a class="sourceLine" id="cb15-11" data-line-number="11"><span class="co">#&gt;  3 RRM2  Comp…      67          1479    1484.         1484.  Up              </span></a>
<a class="sourceLine" id="cb15-12" data-line-number="12"><span class="co">#&gt;  4 FAM1… Comp…     385           569     571.          571.  Up              </span></a>
<a class="sourceLine" id="cb15-13" data-line-number="13"><span class="co">#&gt;  5 YOD1  Comp…     472           470     472.          472.  Up              </span></a>
<a class="sourceLine" id="cb15-14" data-line-number="14"><span class="co">#&gt;  6 ZNF2… Comp…     302          1395    1400.         1400.  Up              </span></a>
<a class="sourceLine" id="cb15-15" data-line-number="15"><span class="co">#&gt;  7 DNAJ… Comp…     557            60      60.2          60.2 Up              </span></a>
<a class="sourceLine" id="cb15-16" data-line-number="16"><span class="co">#&gt;  8 FXN   Comp…     374           382     383.          383.  Up              </span></a>
<a class="sourceLine" id="cb15-17" data-line-number="17"><span class="co">#&gt;  9 NKIR… Comp…       4          1519    1524.         1524.  Up              </span></a>
<a class="sourceLine" id="cb15-18" data-line-number="18"><span class="co">#&gt; 10 NRAS  Comp…       7          2311    2318.         2318.  Up              </span></a>
<a class="sourceLine" id="cb15-19" data-line-number="19"><span class="co">#&gt; # … with 13,422 more rows, and 1 more variable: FC &lt;dbl&gt;</span></a></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">TCGA_E9_A1N5_mirnagene <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> gene_expression, </a>
<a class="sourceLine" id="cb16-3" data-line-number="3">                <span class="dt">miRNA_count =</span> total_read)<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="../reference/calc_perturbation.html">calc_perturbation</a></span>(<span class="dt">node_name=</span> <span class="st">"ACTB"</span>, <span class="dt">cycle=</span><span class="dv">10</span>, <span class="dt">how=</span> <span class="fl">1.87</span>,<span class="dt">limit =</span> <span class="fl">0.1</span>)</a></code></pre></div>
<table class="table">
<thead><tr class="header">
<th><em>perturbation_efficiency</em></th>
<th><em>perturbed_count</em></th>
</tr></thead>
<tbody><tr class="odd">
<td>0.015499</td>
<td>827</td>
</tr></tbody>
</table>
<p>On the other hand, the perturbation eficiency of ATCB gene is higher, when this gene is regulated with 30-fold upregulation like in HIST1H3H.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">TCGA_E9_A1N5_mirnagene <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> gene_expression, </a>
<a class="sourceLine" id="cb17-3" data-line-number="3">                <span class="dt">miRNA_count =</span> total_read)<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="../reference/calc_perturbation.html">calc_perturbation</a></span>(<span class="dt">node_name=</span> <span class="st">"ACTB"</span>, <span class="dt">cycle=</span><span class="dv">10</span>, <span class="dt">how=</span> <span class="dv">30</span>,<span class="dt">limit =</span> <span class="fl">0.1</span>)</a></code></pre></div>
<table class="table">
<thead><tr class="header">
<th><em>perturbation_efficiency</em></th>
<th><em>perturbed_count</em></th>
</tr></thead>
<tbody><tr class="odd">
<td>0.209515</td>
<td>3712</td>
</tr></tbody>
</table>
</div>
<div id="the-sum-of-two-conditions" class="section level2">
<h2 class="hasAnchor">
<a href="#the-sum-of-two-conditions" class="anchor"></a>3.5. The sum of two conditions:</h2>
<p>In a real biological sample, we tested perturbation efficiencies of two genes; * one with low expression but high fold change (HIST1H3H, 30-fold increase in tumor) * another one with high expression but small change in expression level (ACTB, 1.87-fold increase in tumor)</p>
<p>With these two samples, it has been obtained that expression values of genes, rest of the perturbed gene, changed slightly.</p>
<p>Despite high fold change, former gene caused little perturbation. When the perturbation efficiencies of both of these genes are analysed, it has been oberved that HIST1H3H does not affect the other genes in given limit. On the contrary, high expressing gene with very low fold increase in tumor causes greater perturbation in the network. Additionaly, the perturbation efficiency of ACTB gene is quite high from HIST1H3H with 30-fold change, when ACTB is simulated with 30 fold-change.</p>
<p>Thus, if the perturbed node has lower target:total target ratio in group or groups, the efficiency of it can be weak, or vice versa. The efficiency of ACTB gene may be high for this reason, in comparison with HIST1H3H perturbation. In fact, it has been observed that ACTB has not strong perturbation efficiency too. This could be arisen from low miRNA:target ratio or ineffective target nodes which have very low expression levels.</p>
</div>
</div>
<div id="dataset-huge_example-which-includes-mirna-and-gene-expressions-and-mirnatarget-interaction-factors" class="section level1">
<h1 class="hasAnchor">
<a href="#dataset-huge_example-which-includes-mirna-and-gene-expressions-and-mirnatarget-interaction-factors" class="anchor"></a>4. Dataset (<code>huge_example</code>) which includes miRNA and gene expressions and miRNA:target interaction factors</h1>
<div id="description-of-the-huge_example-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#description-of-the-huge_example-dataset" class="anchor"></a>4.1. Description of the <em>huge_example</em> dataset</h2>
<p>Interactions between miRNAs and their targets can be analyzed after the integration of miRNA and targets via various datasets. As an example, we prepared the <em>huge_example</em> dataset. It was generated by integrating:</p>
<ul>
<li>Next-generation RNA sequencing data of a breast cancer patient from TCGA (patient id:TCGA_A7_A0CE)</li>
<li>The microRNA expression values of the same breast cancer patient.</li>
<li>High-throughput miRNA:target determination datasets. We utilised the microRNA:target gene dataset which was integrated from high-throughput experimental studies(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3650559/">CLASH</a> &amp; <a href="https://www.nature.com/articles/ncomms9864">CLEAR-CLiP</a> methods). These datasets give exact microRNA:target pairs because of these methods base on chimeric reading of pairs after binding of microRNAs and their targets. For this reason, the datasets contain detailed information for microRNA:target pairs such as interacted bases (in terms of degree of complementarity) on microRNAs and target mRNAs, location of interaction region (seed) on mRNA and estimated binding energies of pairs.</li>
</ul>
<p>Below, only 6 rows from total of 26,176 rows are shown.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"huge_example"</span>)</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(huge_example)</a>
<a class="sourceLine" id="cb18-3" data-line-number="3"><span class="co">#&gt;   competing           miRNA competing_counts mirnaexpression_normal Energy</span></a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="co">#&gt; 1    TSPAN6     hsa-miR-484             5404              23.058807  -21.4</span></a>
<a class="sourceLine" id="cb18-5" data-line-number="5"><span class="co">#&gt; 2      DPM1  hsa-miR-18b-5p             2472               0.256209  -10.0</span></a>
<a class="sourceLine" id="cb18-6" data-line-number="6"><span class="co">#&gt; 3     SCYL3  hsa-miR-149-5p             1483              34.844420   -5.5</span></a>
<a class="sourceLine" id="cb18-7" data-line-number="7"><span class="co">#&gt; 4     SCYL3  hsa-miR-30a-5p             1483           63031.505507  -13.0</span></a>
<a class="sourceLine" id="cb18-8" data-line-number="8"><span class="co">#&gt; 5  C1orf112 hsa-miR-1296-5p              312               1.793463   -8.1</span></a>
<a class="sourceLine" id="cb18-9" data-line-number="9"><span class="co">#&gt; 6       CFH   hsa-miR-17-5p             5760              89.929349  -17.2</span></a>
<a class="sourceLine" id="cb18-10" data-line-number="10"><span class="co">#&gt;   region_effect seed_type_effect</span></a>
<a class="sourceLine" id="cb18-11" data-line-number="11"><span class="co">#&gt; 1          0.42             0.43</span></a>
<a class="sourceLine" id="cb18-12" data-line-number="12"><span class="co">#&gt; 2          0.84             0.01</span></a>
<a class="sourceLine" id="cb18-13" data-line-number="13"><span class="co">#&gt; 3          0.42             0.01</span></a>
<a class="sourceLine" id="cb18-14" data-line-number="14"><span class="co">#&gt; 4          0.84             0.43</span></a>
<a class="sourceLine" id="cb18-15" data-line-number="15"><span class="co">#&gt; 5          0.42             0.01</span></a>
<a class="sourceLine" id="cb18-16" data-line-number="16"><span class="co">#&gt; 6          0.42             0.01</span></a></code></pre></div>
</div>
<div id="select-a-node-as-trigger" class="section level2">
<h2 class="hasAnchor">
<a href="#select-a-node-as-trigger" class="anchor"></a>4.2. Select a node as trigger</h2>
<p>The node that initiates simulation can be determined according your interest or research.</p>
<p>The dataset, which is a data frame, can be manipulated with <a href="https://www.tidyverse.org/packages/">tidyverse packages</a>. As an example, competing RNAs targeted by less than 5 miRNAs are eliminated to make the network manageable size.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">huge_example <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="st">  </span><span class="kw">add_count</span>(competing) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(n <span class="op">&gt;</span><span class="st"> </span><span class="dv">5</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>n) -&gt;<span class="st"> </span>filtered_example</a>
<a class="sourceLine" id="cb19-5" data-line-number="5"></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(filtered_example)</a>
<a class="sourceLine" id="cb19-7" data-line-number="7"><span class="co">#&gt; # A tibble: 6 x 7</span></a>
<a class="sourceLine" id="cb19-8" data-line-number="8"><span class="co">#&gt;   competing miRNA competing_counts mirnaexpression… Energy region_effect</span></a>
<a class="sourceLine" id="cb19-9" data-line-number="9"><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;            &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;         &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb19-10" data-line-number="10"><span class="co">#&gt; 1 MAD1L1    hsa-…             1909             34.8  -18.6          0.42</span></a>
<a class="sourceLine" id="cb19-11" data-line-number="11"><span class="co">#&gt; 2 MAD1L1    hsa-…             1909            297.   -19.3          0.01</span></a>
<a class="sourceLine" id="cb19-12" data-line-number="12"><span class="co">#&gt; 3 MAD1L1    hsa-…             1909             55.9  -15.4          0.01</span></a>
<a class="sourceLine" id="cb19-13" data-line-number="13"><span class="co">#&gt; 4 MAD1L1    hsa-…             1909             65.8  -13.6          0.42</span></a>
<a class="sourceLine" id="cb19-14" data-line-number="14"><span class="co">#&gt; 5 MAD1L1    hsa-…             1909            603.   -26.7          0.01</span></a>
<a class="sourceLine" id="cb19-15" data-line-number="15"><span class="co">#&gt; 6 MAD1L1    hsa-…             1909           5113.   -30.6          0.01</span></a>
<a class="sourceLine" id="cb19-16" data-line-number="16"><span class="co">#&gt; # … with 1 more variable: seed_type_effect &lt;dbl&gt;</span></a></code></pre></div>
<p>On the other hand, we chose the node <em>GAPDH</em> according to interaction count of the nodes. With the simulation, the graph was visualized after node <em>GAPDH</em> was increased to five fold.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">filtered_example <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> competing_counts, </a>
<a class="sourceLine" id="cb20-3" data-line-number="3">                <span class="dt">miRNA_count =</span> mirnaexpression_normal, </a>
<a class="sourceLine" id="cb20-4" data-line-number="4">                <span class="dt">aff_factor =</span> Energy) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="../reference/update_how.html">update_how</a></span>(<span class="st">"GAPDH"</span>, <span class="dv">5</span>) -&gt;<span class="st"> </span>simulation_GAPDH</a>
<a class="sourceLine" id="cb20-6" data-line-number="6"></a>
<a class="sourceLine" id="cb20-7" data-line-number="7">simulation_GAPDH<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-8" data-line-number="8"><span class="st">  </span><span class="kw"><a href="../reference/vis_graph.html">vis_graph</a></span>(<span class="dt">title =</span> <span class="st">"Distribution of GAPDH gene node"</span>)</a></code></pre></div>
<p><img src="mirtarbase_example_files/figure-html/unnamed-chunk-20-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Let’s visualize each step of simulation via <code><a href="../reference/simulate_vis.html">simulate_vis()</a></code> function.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"></a>
<a class="sourceLine" id="cb21-2" data-line-number="2">simulation_GAPDH<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="../reference/simulate_vis.html">simulate_vis</a></span>(<span class="dt">title =</span> <span class="st">"GAPDH over expression in the real dataset"</span>, <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="co">#&gt; # A tbl_graph: 1526 nodes and 11384 edges</span></a>
<a class="sourceLine" id="cb21-5" data-line-number="5"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="co">#&gt; # A directed acyclic simple graph with 1 component</span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="co">#&gt; # Node Data: 1,526 x 7 (active)</span></a>
<a class="sourceLine" id="cb21-9" data-line-number="9"><span class="co">#&gt;   name    type    node_id initial_count count_pre count_current changes_variable</span></a>
<a class="sourceLine" id="cb21-10" data-line-number="10"><span class="co">#&gt;   &lt;chr&gt;   &lt;chr&gt;     &lt;int&gt;         &lt;dbl&gt;     &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;           </span></a>
<a class="sourceLine" id="cb21-11" data-line-number="11"><span class="co">#&gt; 1 MAD1L1  Compet…       1          1909     1909.         1909. Up              </span></a>
<a class="sourceLine" id="cb21-12" data-line-number="12"><span class="co">#&gt; 2 TFPI    Compet…       2          3377     3377.         3377. Down            </span></a>
<a class="sourceLine" id="cb21-13" data-line-number="13"><span class="co">#&gt; 3 SLC7A2  Compet…       3          8706     8707.         8707. Down            </span></a>
<a class="sourceLine" id="cb21-14" data-line-number="14"><span class="co">#&gt; 4 FKBP4   Compet…       4         20583    20584.        20584. Down            </span></a>
<a class="sourceLine" id="cb21-15" data-line-number="15"><span class="co">#&gt; 5 SLC25A… Compet…       5          1637     1638.         1638. Down            </span></a>
<a class="sourceLine" id="cb21-16" data-line-number="16"><span class="co">#&gt; 6 ST7     Compet…       6          1962     1962.         1962. Up              </span></a>
<a class="sourceLine" id="cb21-17" data-line-number="17"><span class="co">#&gt; # … with 1,520 more rows</span></a>
<a class="sourceLine" id="cb21-18" data-line-number="18"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb21-19" data-line-number="19"><span class="co">#&gt; # Edge Data: 11,384 x 21</span></a>
<a class="sourceLine" id="cb21-20" data-line-number="20"><span class="co">#&gt;    from    to Competing_name miRNA_name competing_counts mirnaexpression… Energy</span></a>
<a class="sourceLine" id="cb21-21" data-line-number="21"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb21-22" data-line-number="22"><span class="co">#&gt; 1     1  1255 MAD1L1         hsa-miR-1…             1909             34.8  -18.6</span></a>
<a class="sourceLine" id="cb21-23" data-line-number="23"><span class="co">#&gt; 2     1  1256 MAD1L1         hsa-miR-1…             1909            297.   -19.3</span></a>
<a class="sourceLine" id="cb21-24" data-line-number="24"><span class="co">#&gt; 3     1  1257 MAD1L1         hsa-miR-1…             1909             55.9  -15.4</span></a>
<a class="sourceLine" id="cb21-25" data-line-number="25"><span class="co">#&gt; # … with 1.138e+04 more rows, and 14 more variables: dummy &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb21-26" data-line-number="26"><span class="co">#&gt; #   afff_factor &lt;dbl&gt;, degg_factor &lt;dbl&gt;, comp_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb21-27" data-line-number="27"><span class="co">#&gt; #   comp_count_pre &lt;dbl&gt;, comp_count_current &lt;dbl&gt;, mirna_count_list &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb21-28" data-line-number="28"><span class="co">#&gt; #   mirna_count_pre &lt;dbl&gt;, mirna_count_current &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb21-29" data-line-number="29"><span class="co">#&gt; #   mirna_count_per_dep &lt;dbl&gt;, effect_current &lt;dbl&gt;, effect_pre &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb21-30" data-line-number="30"><span class="co">#&gt; #   effect_list &lt;list&gt;, mirna_count_per_comp &lt;dbl&gt;</span></a></code></pre></div>
<div class="figure">
<img src="gifs/realexample.gif#center" alt="GAPDH over expression in real dataset"><p class="caption">GAPDH over expression in real dataset</p>
</div>
<p>Now, we can track changes in expression levels at every node for 3 cycles when GAPDH is overexpressed 5-fold.</p>
<ul>
<li><p>After increase in GAPDH expression level in the first graph, the responses of the other competing elements to the GAPDH distributions were calculated.</p></li>
<li><p>The changing regulations (up or down) were observed as a result of interactions in the second graph.</p></li>
<li><p>When three graphs were carefully compared to each other, it can be observed that the expression levels of nodes change continuously at each stage.</p></li>
</ul>
</div>
</div>
<div id="finding-perturbation-efficiency-on-an-experimental-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#finding-perturbation-efficiency-on-an-experimental-dataset" class="anchor"></a>5. Finding perturbation efficiency on an experimental dataset</h1>
<p><code><a href="../reference/find_node_perturbation.html">find_node_perturbation()</a></code> runs <code>calc_perturb</code> on all nodes in the network in parallel with help of the <code>future</code> and <code>furrr</code> packages. In this vignette, the function is demonstrated on the <code>midsamp</code> data. This dataset is not comparable to actual biological miRNA:target gene datasets in size and complexity. Although <code><a href="../reference/find_node_perturbation.html">find_node_perturbation()</a></code> runs in parallel it might take long time to run in real huge biological datasets.</p>
<p>In real biological datasets, more complex interactions whether functional or non-functional could be observed. We have improved our approach with <code>fast</code> argument in <code><a href="../reference/find_node_perturbation.html">find_node_perturbation()</a></code> based on selection of elements that could be affected from perturbation. In this fucntion, <code>fast</code> argument specifies the percentage of the competing amount that can be affected within the initial competing amount and acts as a selection parameter. For instance, in huge_example data:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"></a>
<a class="sourceLine" id="cb22-2" data-line-number="2">filtered_example<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="../reference/priming_graph.html">priming_graph</a></span>(<span class="dt">competing_count =</span> competing_counts, <span class="dt">miRNA_count =</span> mirnaexpression_normal)<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="../reference/find_node_perturbation.html">find_node_perturbation</a></span>(<span class="dt">how=</span><span class="dv">5</span>, <span class="dt">cycle=</span><span class="dv">3</span>, <span class="dt">fast =</span> <span class="dv">10</span>)<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-5" data-line-number="5"><span class="st">  </span><span class="kw">select</span>(name, perturbation_efficiency, perturbed_count) -&gt;<span class="st"> </span>entire_perturbation</a>
<a class="sourceLine" id="cb22-6" data-line-number="6"><span class="co">#&gt; Warning in priming_graph(., competing_count = competing_counts, miRNA_count = mirnaexpression_normal): First column is processed as competing and the second as miRNA.</span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="co">#&gt; Warning: `as_quosure()` requires an explicit environment as of rlang 0.3.0.</span></a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="co">#&gt; Please supply `env`.</span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="co">#&gt; This warning is displayed once per session.</span></a>
<a class="sourceLine" id="cb22-10" data-line-number="10"><span class="co">#&gt; Subsetting by edges</span></a>
<a class="sourceLine" id="cb22-11" data-line-number="11"><span class="co">#&gt; Warning: `cols` is now required.</span></a>
<a class="sourceLine" id="cb22-12" data-line-number="12"><span class="co">#&gt; Please use `cols = c(eff_count)`</span></a></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"></a>
<a class="sourceLine" id="cb23-2" data-line-number="2">entire_perturbation<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(perturbation_efficiency), <span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(perturbed_count))<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4"><span class="st">  </span><span class="kw">select</span>(name, perturbation_efficiency, perturbed_count)</a>
<a class="sourceLine" id="cb23-5" data-line-number="5"><span class="co">#&gt; # A tibble: 223 x 3</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="co">#&gt;    name    perturbation_efficiency perturbed_count</span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="co">#&gt;    &lt;chr&gt;                     &lt;dbl&gt;           &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="co">#&gt;  1 SLC25A5               0.320                 207</span></a>
<a class="sourceLine" id="cb23-9" data-line-number="9"><span class="co">#&gt;  2 MSL3                  0.0696                207</span></a>
<a class="sourceLine" id="cb23-10" data-line-number="10"><span class="co">#&gt;  3 MYCBP2                0.0695                205</span></a>
<a class="sourceLine" id="cb23-11" data-line-number="11"><span class="co">#&gt;  4 VPS41                 0.0472                205</span></a>
<a class="sourceLine" id="cb23-12" data-line-number="12"><span class="co">#&gt;  5 ADIPOR2               0.175                 207</span></a>
<a class="sourceLine" id="cb23-13" data-line-number="13"><span class="co">#&gt;  6 SCMH1                 0.0879                207</span></a>
<a class="sourceLine" id="cb23-14" data-line-number="14"><span class="co">#&gt;  7 NR1H4                 0.0000227             207</span></a>
<a class="sourceLine" id="cb23-15" data-line-number="15"><span class="co">#&gt;  8 RALBP1                0.156                 207</span></a>
<a class="sourceLine" id="cb23-16" data-line-number="16"><span class="co">#&gt;  9 VCL                   0.388                 207</span></a>
<a class="sourceLine" id="cb23-17" data-line-number="17"><span class="co">#&gt; 10 CTNNA1                0.683                 207</span></a>
<a class="sourceLine" id="cb23-18" data-line-number="18"><span class="co">#&gt; # … with 213 more rows</span></a></code></pre></div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>6. Session Info</h1>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="co">#&gt; R Under development (unstable) (2019-11-23 r77455)</span></a>
<a class="sourceLine" id="cb24-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb24-4" data-line-number="4"><span class="co">#&gt; Running under: Ubuntu 16.04.6 LTS</span></a>
<a class="sourceLine" id="cb24-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb24-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb24-7" data-line-number="7"><span class="co">#&gt; BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so</span></a>
<a class="sourceLine" id="cb24-8" data-line-number="8"><span class="co">#&gt; LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so</span></a>
<a class="sourceLine" id="cb24-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb24-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb24-11" data-line-number="11"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></a>
<a class="sourceLine" id="cb24-12" data-line-number="12"><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></a>
<a class="sourceLine" id="cb24-13" data-line-number="13"><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></a>
<a class="sourceLine" id="cb24-14" data-line-number="14"><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></a>
<a class="sourceLine" id="cb24-15" data-line-number="15"><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></a>
<a class="sourceLine" id="cb24-16" data-line-number="16"><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb24-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb24-18" data-line-number="18"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb24-19" data-line-number="19"><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></a>
<a class="sourceLine" id="cb24-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb24-21" data-line-number="21"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb24-22" data-line-number="22"><span class="co">#&gt; [1] ceRNAnetsim_0.99.16 tidygraph_1.1.2     dplyr_0.8.3        </span></a>
<a class="sourceLine" id="cb24-23" data-line-number="23"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb24-24" data-line-number="24"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb24-25" data-line-number="25"><span class="co">#&gt;  [1] tidyselect_0.2.5   xfun_0.11          purrr_0.3.3        listenv_0.7.0     </span></a>
<a class="sourceLine" id="cb24-26" data-line-number="26"><span class="co">#&gt;  [5] graphlayouts_0.5.0 colorspace_1.4-1   vctrs_0.2.0        htmltools_0.4.0   </span></a>
<a class="sourceLine" id="cb24-27" data-line-number="27"><span class="co">#&gt;  [9] viridisLite_0.3.0  yaml_2.2.0         utf8_1.1.4         rlang_0.4.2       </span></a>
<a class="sourceLine" id="cb24-28" data-line-number="28"><span class="co">#&gt; [13] pkgdown_1.4.1      pillar_1.4.2       glue_1.3.1         tweenr_1.0.1      </span></a>
<a class="sourceLine" id="cb24-29" data-line-number="29"><span class="co">#&gt; [17] lifecycle_0.1.0    stringr_1.4.0      munsell_0.5.0      gtable_0.3.0      </span></a>
<a class="sourceLine" id="cb24-30" data-line-number="30"><span class="co">#&gt; [21] future_1.15.0      codetools_0.2-16   memoise_1.1.0      evaluate_0.14     </span></a>
<a class="sourceLine" id="cb24-31" data-line-number="31"><span class="co">#&gt; [25] labeling_0.3       knitr_1.26         parallel_4.0.0     fansi_0.4.0       </span></a>
<a class="sourceLine" id="cb24-32" data-line-number="32"><span class="co">#&gt; [29] furrr_0.1.0        Rcpp_1.0.3         backports_1.1.5    scales_1.1.0      </span></a>
<a class="sourceLine" id="cb24-33" data-line-number="33"><span class="co">#&gt; [33] desc_1.2.0         farver_2.0.1       fs_1.3.1           gridExtra_2.3     </span></a>
<a class="sourceLine" id="cb24-34" data-line-number="34"><span class="co">#&gt; [37] ggforce_0.3.1      ggplot2_3.2.1      digest_0.6.23      stringi_1.4.3     </span></a>
<a class="sourceLine" id="cb24-35" data-line-number="35"><span class="co">#&gt; [41] ggrepel_0.8.1      polyclip_1.10-0    rprojroot_1.3-2    grid_4.0.0        </span></a>
<a class="sourceLine" id="cb24-36" data-line-number="36"><span class="co">#&gt; [45] cli_1.1.0          tools_4.0.0        magrittr_1.5       lazyeval_0.2.2    </span></a>
<a class="sourceLine" id="cb24-37" data-line-number="37"><span class="co">#&gt; [49] tibble_2.1.3       ggraph_2.0.0       crayon_1.3.4       tidyr_1.0.0       </span></a>
<a class="sourceLine" id="cb24-38" data-line-number="38"><span class="co">#&gt; [53] pkgconfig_2.0.3    zeallot_0.1.0      MASS_7.3-51.4      assertthat_0.2.1  </span></a>
<a class="sourceLine" id="cb24-39" data-line-number="39"><span class="co">#&gt; [57] rmarkdown_1.15     viridis_0.5.1      R6_2.4.1           globals_0.12.4    </span></a>
<a class="sourceLine" id="cb24-40" data-line-number="40"><span class="co">#&gt; [61] igraph_1.2.4.1     compiler_4.0.0</span></a></code></pre></div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-colaprico2015tcgabiolinks">
<p>Colaprico, Antonio, Tiago C Silva, Catharina Olsen, Luciano Garofano, Claudia Cava, Davide Garolini, Thais S Sabedot, et al. 2015. “TCGAbiolinks: An R/Bioconductor Package for Integrative Analysis of Tcga Data.” <em>Nucleic Acids Research</em> 44 (8). Oxford University Press: e71–e71.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">1. Introduction</a></li>
      <li><a href="#installation">2. Installation</a></li>
      <li>
<a href="#integration-of-dataset-which-includes-only-mirna-and-gene-expression-values">3. Integration of dataset which includes only miRNA and gene expression values</a><ul class="nav nav-pills nav-stacked">
<li><a href="#mirnatarget-pairs">3.1. miRNA:target pairs</a></li>
      <li><a href="#gene-expression-in-normal-and-tumor-samples">3.2. Gene expression in normal and tumor samples</a></li>
      <li><a href="#mirna-expression-data">3.3. miRNA expression data</a></li>
      <li><a href="#integrating-and-analysing-data">3.4. Integrating and analysing data</a></li>
      <li><a href="#the-sum-of-two-conditions">3.5. The sum of two conditions:</a></li>
      </ul>
</li>
      <li>
<a href="#dataset-huge_example-which-includes-mirna-and-gene-expressions-and-mirnatarget-interaction-factors">4. Dataset (<code>huge_example</code>) which includes miRNA and gene expressions and miRNA:target interaction factors</a><ul class="nav nav-pills nav-stacked">
<li><a href="#description-of-the-huge_example-dataset">4.1. Description of the <em>huge_example</em> dataset</a></li>
      <li><a href="#select-a-node-as-trigger">4.2. Select a node as trigger</a></li>
      </ul>
</li>
      <li><a href="#finding-perturbation-efficiency-on-an-experimental-dataset">5. Finding perturbation efficiency on an experimental dataset</a></li>
      <li>
<a href="#session-info">6. Session Info</a><ul class="nav nav-pills nav-stacked">
<li><a href="#references">References</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Selcen Ari Yuka, Alper Yilmaz.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'fbd8d79ed49f21fc9ca7b2cf601a3f5b',
    indexName: 'ceRNAnetsim',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
